## R MS Data

### mzR {-}

The `mzR` package is a powerful tool for analyzing large raw data files in proteomics research. It offers a direct interface to the [proteowizard](http://proteowizard.sourceforge.net/) code base, which is a widely-used software framework for mass spectrometry data analysis. By leveraging a substantial proportion of *pwiz*'s C/C++ code, `mzR` is able to provide fast and efficient parsing of these complex data files.

\

+---------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| ![](images/05.png){height="68"} | **Docs** | [tutorial](https://lgatto.github.io/2020-02-17-RProt-Berlin/raw-ms-data-mzr-and-msnbase.html), [PDF](https://bioconductor.org/packages/release/bioc/manuals/mzR/man/mzR.pdf) |
|                                 |                                                                                                                                                                                         |
|                                 | **Code** | [Bioconductor](https://bioconductor.org/packages/release/bioc/html/mzR.html)                                                                                                 |
|                                 |                                                                                                                                                                                         |
|                                 | **Cite** | *Nat Biotech* 2012, 30(10), 918–920                                                                                                                                          |
+---------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+

\

```{r, eval=FALSE}
BiocManager::install("MSnbase")
```

In addition to its parsing capabilities, `mzR` includes a range of useful functions for working with mass spectrometry data. This package supports a wide variety of file formats, and provides functions for reading, writing, and manipulating data in these formats. Furthermore, `mzR` includes features for working with tandem mass spectrometry (MS/MS) data, including functions for spectral processing and peak picking.

### MSnbase {-}

Often, the full access to raw data provides a complete sense of control. However, there are times when attempting to achieve mundane tasks can be extremely tedious and repetitive. This is where the concept of abstraction comes in. It is unnecessary to be aware of every detail that is exposed by mzR to work with and manipulate raw data. This is where `MSnbase` and the `MSnExp` data structure come in. These tools provide a smoother and more efficient approach to handle and annotate raw data.

\

+---------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| ![](images/05.png){height="68"} | **Docs** | [tutorial](https://bioconductor.org/packages/release/bioc/vignettes/MSnbase/inst/doc/v05-MSnbase-development.html), [PDF](https://bioconductor.org/packages/release/bioc/manuals/MSnbase/man/MSnbase.pdf) |
|                                 |                                                                                                                                                                                                                      |
|                                 | **Code** | [Bioconductor](https://bioconductor.org/packages/release/bioc/html/MSnbase.html)                                                                                                                          |
|                                 |                                                                                                                                                                                                                      |
|                                 | **Cite** | *Bioinformatics* 2012, 28, 288-289.                                                                                                                                                                       |
+---------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+

\

```{r, eval=FALSE}
BiocManager::install("MSnbase")
```

### xcms {-}

XCMS is an R package that provides a collection of functions for processing and analyzing mass spectrometry (MS) data. This package is designed to extract meaningful information from large data sets generated by MS-based experiments, such as metabolomics and proteomics studies.

\

+---------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| ![](images/05.png){height="68"} | **Docs** | [tutorial](https://bioconductor.org/packages/release/bioc/vignettes/xcms/inst/doc/xcms-lcms-ms.html), [PDF](https://bioconductor.org/packages/3.12/bioc/manuals/xcms/man/xcms.pdf) |
|                                 |                                                                                                                                                                                               |
|                                 | **Code** | [Bioconductor](https://bioconductor.org/packages/3.12/bioc/html/xcms.html)                                                                                                         |
|                                 |                                                                                                                                                                                               |
|                                 | **Cite** | *Analytical Chemistry* 2008, 78, 779–787.                                                                                                                                          |
+---------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+

\

```{r, eval=FALSE}
BiocManager::install("xcms")
```

XCMS offers a wide range of functionalities that are necessary for preprocessing and analyzing raw MS data. Some of the key features of XCMS include:

- Data preprocessing: The package offers a variety of preprocessing functions, such as peak detection, retention time correction, and noise filtering.
- Feature detection: XCMS provides functions for detecting and aligning features across multiple samples, which is essential for comparative analysis.
- Statistical analysis: The package also includes functions for statistical analysis, such as principal component analysis (PCA), hierarchical clustering, and differential expression analysis.
- Visualization: XCMS provides several visualization tools for exploring and interpreting MS data, including heatmaps, scatter plots, and volcano plots.

One of the main advantages of XCMS is its flexibility and scalability. This package is highly customizable, allowing users to tailor the analysis pipeline to their specific needs. In addition, XCMS is designed to handle large data sets, making it suitable for high-throughput experiments.
