<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Mass Spectrometry R Packages (work in progress) | Introduction to R for Mass Spectrometrists</title>
  <meta name="description" content="An introductory approach to using R in the analysis of mass spectrometry related data." />
  <meta name="generator" content="bookdown 0.33 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Mass Spectrometry R Packages (work in progress) | Introduction to R for Mass Spectrometrists" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="An introductory approach to using R in the analysis of mass spectrometry related data." />
  <meta name="github-repo" content="jeffsocal/ASMS_R_Basics" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Mass Spectrometry R Packages (work in progress) | Introduction to R for Mass Spectrometrists" />
  
  <meta name="twitter:description" content="An introductory approach to using R in the analysis of mass spectrometry related data." />
  

<meta name="author" content="Jeff Jones, Heath Patterson, Ryan Benz" />


<meta name="date" content="2023-05-02" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="mass-spectrometry-data.html"/>
<link rel="next" href="sharing.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Introduction to R</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="" data-path="acknowledgements.html"><a href="acknowledgements.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="1" data-path="introduction-to-r.html"><a href="introduction-to-r.html"><i class="fa fa-check"></i><b>1</b> Introduction to R</a>
<ul>
<li class="chapter" data-level="" data-path="introduction-to-r.html"><a href="introduction-to-r.html#why-choose-it"><i class="fa fa-check"></i>Why choose it?</a></li>
<li class="chapter" data-level="" data-path="introduction-to-r.html"><a href="introduction-to-r.html#what-you-can-do-with-it"><i class="fa fa-check"></i>What you can do with it?</a></li>
<li class="chapter" data-level="" data-path="introduction-to-r.html"><a href="introduction-to-r.html#the-r-learning-curve"><i class="fa fa-check"></i>The R Learning Curve</a></li>
<li class="chapter" data-level="" data-path="introduction-to-r.html"><a href="introduction-to-r.html#thoughts-about-learning-r-and-how-to-code"><i class="fa fa-check"></i>Thoughts about learning R and how to code</a></li>
<li class="chapter" data-level="" data-path="introduction-to-r.html"><a href="introduction-to-r.html#alternatives"><i class="fa fa-check"></i>Alternatives</a>
<ul>
<li class="chapter" data-level="" data-path="introduction-to-r.html"><a href="introduction-to-r.html#did-you-know"><i class="fa fa-check"></i>Did you know</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="installation.html"><a href="installation.html"><i class="fa fa-check"></i><b>2</b> Installation</a>
<ul>
<li class="chapter" data-level="2.1" data-path="installation.html"><a href="installation.html#r-interpreter"><i class="fa fa-check"></i><b>2.1</b> R interpreter</a></li>
<li class="chapter" data-level="2.2" data-path="installation.html"><a href="installation.html#rstudio"><i class="fa fa-check"></i><b>2.2</b> Rstudio</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="installation.html"><a href="installation.html#ide-layout"><i class="fa fa-check"></i><b>2.2.1</b> IDE Layout</a></li>
<li class="chapter" data-level="2.2.2" data-path="installation.html"><a href="installation.html#usage-..-running-lines-of-code-in-rstudio"><i class="fa fa-check"></i><b>2.2.2</b> Usage .. Running lines of code in RStudio</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="installation.html"><a href="installation.html#packages"><i class="fa fa-check"></i><b>2.3</b> Packages</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="installation.html"><a href="installation.html#what-are-r-packages"><i class="fa fa-check"></i><b>2.3.1</b> What are R Packages?</a></li>
<li class="chapter" data-level="2.3.2" data-path="installation.html"><a href="installation.html#where-to-get-r-packages"><i class="fa fa-check"></i><b>2.3.2</b> Where to get R Packages</a></li>
<li class="chapter" data-level="2.3.3" data-path="installation.html"><a href="installation.html#installing-r-packages"><i class="fa fa-check"></i><b>2.3.3</b> Installing R Packages</a></li>
<li class="chapter" data-level="2.3.4" data-path="installation.html"><a href="installation.html#potential-gotchas"><i class="fa fa-check"></i><b>2.3.4</b> Potential Gotchas</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="installation.html"><a href="installation.html#packages-utilized-in-this-book"><i class="fa fa-check"></i><b>2.4</b> Packages Utilized in This Book</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="installation.html"><a href="installation.html#tidyverse"><i class="fa fa-check"></i><b>2.4.1</b> tidyverse</a></li>
<li class="chapter" data-level="2.4.2" data-path="installation.html"><a href="installation.html#mass-spectrometry-specific-packages"><i class="fa fa-check"></i><b>2.4.2</b> Mass Spectrometry Specific Packages</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="the-basics.html"><a href="the-basics.html"><i class="fa fa-check"></i><b>3</b> The Basics</a>
<ul>
<li class="chapter" data-level="3.1" data-path="the-basics.html"><a href="the-basics.html#reserved-words"><i class="fa fa-check"></i><b>3.1</b> Reserved Words</a></li>
<li class="chapter" data-level="3.2" data-path="the-basics.html"><a href="the-basics.html#syntax"><i class="fa fa-check"></i><b>3.2</b> Syntax</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="the-basics.html"><a href="the-basics.html#comments"><i class="fa fa-check"></i><b>3.2.1</b> Comments</a></li>
<li class="chapter" data-level="3.2.2" data-path="the-basics.html"><a href="the-basics.html#strings"><i class="fa fa-check"></i><b>3.2.2</b> Strings</a></li>
<li class="chapter" data-level="3.2.3" data-path="the-basics.html"><a href="the-basics.html#numbers"><i class="fa fa-check"></i><b>3.2.3</b> Numbers</a></li>
<li class="chapter" data-level="3.2.4" data-path="the-basics.html"><a href="the-basics.html#operators"><i class="fa fa-check"></i><b>3.2.4</b> Operators</a></li>
<li class="chapter" data-level="3.2.5" data-path="the-basics.html"><a href="the-basics.html#variables"><i class="fa fa-check"></i><b>3.2.5</b> Variables</a></li>
<li class="chapter" data-level="3.2.6" data-path="the-basics.html"><a href="the-basics.html#statements"><i class="fa fa-check"></i><b>3.2.6</b> Statements</a></li>
<li class="chapter" data-level="3.2.7" data-path="the-basics.html"><a href="the-basics.html#functions"><i class="fa fa-check"></i><b>3.2.7</b> Functions</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="the-basics.html"><a href="the-basics.html#flow-control"><i class="fa fa-check"></i><b>3.3</b> Flow-Control</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="the-basics.html"><a href="the-basics.html#if-else-statements"><i class="fa fa-check"></i><b>3.3.1</b> If-Else Statements</a></li>
<li class="chapter" data-level="3.3.2" data-path="the-basics.html"><a href="the-basics.html#loops"><i class="fa fa-check"></i><b>3.3.2</b> Loops</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="the-basics.html"><a href="the-basics.html#projects"><i class="fa fa-check"></i><b>3.4</b> Projects</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="the-basics.html"><a href="the-basics.html#creating"><i class="fa fa-check"></i><b>3.4.1</b> Creating</a></li>
<li class="chapter" data-level="3.4.2" data-path="the-basics.html"><a href="the-basics.html#editing"><i class="fa fa-check"></i><b>3.4.2</b> Editing</a></li>
<li class="chapter" data-level="3.4.3" data-path="the-basics.html"><a href="the-basics.html#organizing"><i class="fa fa-check"></i><b>3.4.3</b> Organizing</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="the-basics.html"><a href="the-basics.html#exercises"><i class="fa fa-check"></i>Exercises</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="data-structures.html"><a href="data-structures.html"><i class="fa fa-check"></i><b>4</b> Data Structures</a>
<ul>
<li class="chapter" data-level="4.1" data-path="data-structures.html"><a href="data-structures.html#variable"><i class="fa fa-check"></i><b>4.1</b> Variable</a></li>
<li class="chapter" data-level="4.2" data-path="data-structures.html"><a href="data-structures.html#vector"><i class="fa fa-check"></i><b>4.2</b> Vector</a></li>
<li class="chapter" data-level="4.3" data-path="data-structures.html"><a href="data-structures.html#matrix"><i class="fa fa-check"></i><b>4.3</b> Matrix</a></li>
<li class="chapter" data-level="4.4" data-path="data-structures.html"><a href="data-structures.html#data-frame"><i class="fa fa-check"></i><b>4.4</b> Data Frame</a></li>
<li class="chapter" data-level="4.5" data-path="data-structures.html"><a href="data-structures.html#list"><i class="fa fa-check"></i><b>4.5</b> List</a></li>
<li class="chapter" data-level="4.6" data-path="data-structures.html"><a href="data-structures.html#how-to-tell-what-you-are-dealing-with"><i class="fa fa-check"></i><b>4.6</b> How to tell what you are dealing with</a></li>
<li class="chapter" data-level="" data-path="data-structures.html"><a href="data-structures.html#exercises-1"><i class="fa fa-check"></i>Exercises</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="data-tables.html"><a href="data-tables.html"><i class="fa fa-check"></i><b>5</b> Data Tables</a>
<ul>
<li class="chapter" data-level="5.1" data-path="data-tables.html"><a href="data-tables.html#tibble-a-new-data-structure"><i class="fa fa-check"></i><b>5.1</b> Tibble, a new Data Structure</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="data-wrangling.html"><a href="data-wrangling.html"><i class="fa fa-check"></i><b>6</b> Data Wrangling</a>
<ul>
<li class="chapter" data-level="6.1" data-path="data-wrangling.html"><a href="data-wrangling.html#tidy-data"><i class="fa fa-check"></i><b>6.1</b> Tidy Data</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="data-visualization.html"><a href="data-visualization.html"><i class="fa fa-check"></i><b>7</b> Data Visualization</a>
<ul>
<li class="chapter" data-level="7.1" data-path="data-visualization.html"><a href="data-visualization.html#base-plot"><i class="fa fa-check"></i><b>7.1</b> Base: plot</a></li>
<li class="chapter" data-level="7.2" data-path="data-visualization.html"><a href="data-visualization.html#ggplot"><i class="fa fa-check"></i><b>7.2</b> GGPlot</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="data-visualization.html"><a href="data-visualization.html#syntax-1"><i class="fa fa-check"></i><b>7.2.1</b> Syntax</a></li>
<li class="chapter" data-level="7.2.2" data-path="data-visualization.html"><a href="data-visualization.html#basic-data-plotting"><i class="fa fa-check"></i><b>7.2.2</b> Basic Data Plotting</a></li>
<li class="chapter" data-level="7.2.3" data-path="data-visualization.html"><a href="data-visualization.html#data-distributions"><i class="fa fa-check"></i><b>7.2.3</b> Data Distributions</a></li>
<li class="chapter" data-level="7.2.4" data-path="data-visualization.html"><a href="data-visualization.html#extended-syntax"><i class="fa fa-check"></i><b>7.2.4</b> Extended Syntax</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="data-visualization.html"><a href="data-visualization.html#plotly"><i class="fa fa-check"></i><b>7.3</b> Plotly</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="mass-spectrometry-data.html"><a href="mass-spectrometry-data.html"><i class="fa fa-check"></i><b>8</b> Mass Spectrometry Data</a>
<ul>
<li class="chapter" data-level="8.1" data-path="mass-spectrometry-data.html"><a href="mass-spectrometry-data.html#commercial"><i class="fa fa-check"></i><b>8.1</b> Commercial</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="mass-spectrometry-data.html"><a href="mass-spectrometry-data.html#raw-thermo"><i class="fa fa-check"></i><b>8.1.1</b> RAW (Thermo)</a></li>
<li class="chapter" data-level="8.1.2" data-path="mass-spectrometry-data.html"><a href="mass-spectrometry-data.html#wiff-sciex"><i class="fa fa-check"></i><b>8.1.2</b> WIFF (Sciex)</a></li>
<li class="chapter" data-level="8.1.3" data-path="mass-spectrometry-data.html"><a href="mass-spectrometry-data.html#d-agilent"><i class="fa fa-check"></i><b>8.1.3</b> D (Agilent)</a></li>
<li class="chapter" data-level="8.1.4" data-path="mass-spectrometry-data.html"><a href="mass-spectrometry-data.html#d-bruker"><i class="fa fa-check"></i><b>8.1.4</b> D (Bruker)</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="mass-spectrometry-data.html"><a href="mass-spectrometry-data.html#open-access-raw-data"><i class="fa fa-check"></i><b>8.2</b> Open Access Raw Data</a>
<ul>
<li class="chapter" data-level="" data-path="mass-spectrometry-data.html"><a href="mass-spectrometry-data.html#conversion-tools"><i class="fa fa-check"></i>Conversion Tools</a></li>
<li class="chapter" data-level="" data-path="mass-spectrometry-data.html"><a href="mass-spectrometry-data.html#mzxml"><i class="fa fa-check"></i>mzXML</a></li>
<li class="chapter" data-level="" data-path="mass-spectrometry-data.html"><a href="mass-spectrometry-data.html#mzml"><i class="fa fa-check"></i>mzML</a></li>
<li class="chapter" data-level="" data-path="mass-spectrometry-data.html"><a href="mass-spectrometry-data.html#mzmlb"><i class="fa fa-check"></i>mzMLb</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="mass-spectrometry-r-packages-work-in-progress.html"><a href="mass-spectrometry-r-packages-work-in-progress.html"><i class="fa fa-check"></i><b>9</b> Mass Spectrometry R Packages (work in progress)</a>
<ul>
<li class="chapter" data-level="9.1" data-path="mass-spectrometry-r-packages-work-in-progress.html"><a href="mass-spectrometry-r-packages-work-in-progress.html#xcms"><i class="fa fa-check"></i><b>9.1</b> xcms</a></li>
<li class="chapter" data-level="9.2" data-path="mass-spectrometry-r-packages-work-in-progress.html"><a href="mass-spectrometry-r-packages-work-in-progress.html#deqms"><i class="fa fa-check"></i><b>9.2</b> deqms</a></li>
<li class="chapter" data-level="9.3" data-path="mass-spectrometry-r-packages-work-in-progress.html"><a href="mass-spectrometry-r-packages-work-in-progress.html#msempire"><i class="fa fa-check"></i><b>9.3</b> msempire</a></li>
<li class="chapter" data-level="9.4" data-path="mass-spectrometry-r-packages-work-in-progress.html"><a href="mass-spectrometry-r-packages-work-in-progress.html#msqrob"><i class="fa fa-check"></i><b>9.4</b> msqrob</a></li>
<li class="chapter" data-level="9.5" data-path="mass-spectrometry-r-packages-work-in-progress.html"><a href="mass-spectrometry-r-packages-work-in-progress.html#msstatstmt"><i class="fa fa-check"></i><b>9.5</b> MSstatsTMT</a></li>
<li class="chapter" data-level="9.6" data-path="mass-spectrometry-r-packages-work-in-progress.html"><a href="mass-spectrometry-r-packages-work-in-progress.html#msstats"><i class="fa fa-check"></i><b>9.6</b> MSstats</a>
<ul>
<li class="chapter" data-level="9.6.1" data-path="mass-spectrometry-r-packages-work-in-progress.html"><a href="mass-spectrometry-r-packages-work-in-progress.html#installation-1"><i class="fa fa-check"></i><b>9.6.1</b> Installation</a></li>
<li class="chapter" data-level="9.6.2" data-path="mass-spectrometry-r-packages-work-in-progress.html"><a href="mass-spectrometry-r-packages-work-in-progress.html#loading-data"><i class="fa fa-check"></i><b>9.6.2</b> Loading Data</a></li>
<li class="chapter" data-level="9.6.3" data-path="mass-spectrometry-r-packages-work-in-progress.html"><a href="mass-spectrometry-r-packages-work-in-progress.html#quality-control"><i class="fa fa-check"></i><b>9.6.3</b> Quality Control</a></li>
<li class="chapter" data-level="9.6.4" data-path="mass-spectrometry-r-packages-work-in-progress.html"><a href="mass-spectrometry-r-packages-work-in-progress.html#differential-expression-analysis"><i class="fa fa-check"></i><b>9.6.4</b> Differential Expression Analysis</a></li>
<li class="chapter" data-level="9.6.5" data-path="mass-spectrometry-r-packages-work-in-progress.html"><a href="mass-spectrometry-r-packages-work-in-progress.html#conclusion"><i class="fa fa-check"></i><b>9.6.5</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="9.7" data-path="mass-spectrometry-r-packages-work-in-progress.html"><a href="mass-spectrometry-r-packages-work-in-progress.html#msgrob2"><i class="fa fa-check"></i><b>9.7</b> msgrob2</a>
<ul>
<li class="chapter" data-level="9.7.1" data-path="mass-spectrometry-r-packages-work-in-progress.html"><a href="mass-spectrometry-r-packages-work-in-progress.html#code-examples"><i class="fa fa-check"></i><b>9.7.1</b> Code Examples</a></li>
</ul></li>
<li class="chapter" data-level="9.8" data-path="mass-spectrometry-r-packages-work-in-progress.html"><a href="mass-spectrometry-r-packages-work-in-progress.html#statspro-r-package"><i class="fa fa-check"></i><b>9.8</b> StatsPro R Package</a></li>
<li class="chapter" data-level="9.9" data-path="mass-spectrometry-r-packages-work-in-progress.html"><a href="mass-spectrometry-r-packages-work-in-progress.html#tidyproteomics"><i class="fa fa-check"></i><b>9.9</b> Tidyproteomics</a>
<ul>
<li class="chapter" data-level="9.9.1" data-path="mass-spectrometry-r-packages-work-in-progress.html"><a href="mass-spectrometry-r-packages-work-in-progress.html#loading-data-1"><i class="fa fa-check"></i><b>9.9.1</b> Loading Data</a></li>
<li class="chapter" data-level="9.9.2" data-path="mass-spectrometry-r-packages-work-in-progress.html"><a href="mass-spectrometry-r-packages-work-in-progress.html#data-summaries"><i class="fa fa-check"></i><b>9.9.2</b> Data Summaries</a></li>
<li class="chapter" data-level="9.9.3" data-path="mass-spectrometry-r-packages-work-in-progress.html"><a href="mass-spectrometry-r-packages-work-in-progress.html#normalization-and-imputation"><i class="fa fa-check"></i><b>9.9.3</b> Normalization and Imputation</a></li>
<li class="chapter" data-level="9.9.4" data-path="mass-spectrometry-r-packages-work-in-progress.html"><a href="mass-spectrometry-r-packages-work-in-progress.html#expression-analysis"><i class="fa fa-check"></i><b>9.9.4</b> Expression Analysis</a></li>
<li class="chapter" data-level="9.9.5" data-path="mass-spectrometry-r-packages-work-in-progress.html"><a href="mass-spectrometry-r-packages-work-in-progress.html#advanced"><i class="fa fa-check"></i><b>9.9.5</b> Advanced</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="sharing.html"><a href="sharing.html"><i class="fa fa-check"></i><b>10</b> Sharing</a>
<ul>
<li class="chapter" data-level="10.1" data-path="sharing.html"><a href="sharing.html#packrat"><i class="fa fa-check"></i><b>10.1</b> Packrat</a>
<ul>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#initiating-a-packrat-project"><i class="fa fa-check"></i>Initiating a Packrat Project</a></li>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#installing-a-package-into-the-project-library"><i class="fa fa-check"></i>Installing a Package into the Project Library</a></li>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#loading-a-package-from-the-project-library"><i class="fa fa-check"></i>Loading a Package from the Project Library</a></li>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#updating-a-package-in-the-project-library"><i class="fa fa-check"></i>Updating a Package in the Project Library</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="sharing.html"><a href="sharing.html#notebooks"><i class="fa fa-check"></i><b>10.2</b> Notebooks</a>
<ul>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#using-notebooks"><i class="fa fa-check"></i>Using Notebooks</a></li>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#sharing-a-notebook"><i class="fa fa-check"></i>Sharing A Notebook</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="sharing.html"><a href="sharing.html#github"><i class="fa fa-check"></i><b>10.3</b> GitHub</a>
<ul>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#how-to-use-github"><i class="fa fa-check"></i>How to Use GitHub</a></li>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#how-to-share-a-github-repository"><i class="fa fa-check"></i>How to Share A GitHub Repository</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="sharing.html"><a href="sharing.html#docker"><i class="fa fa-check"></i><b>10.4</b> Docker</a></li>
<li class="chapter" data-level="10.5" data-path="sharing.html"><a href="sharing.html#r-packages"><i class="fa fa-check"></i><b>10.5</b> R Packages</a>
<ul>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#creating-a-package-in-rstudio"><i class="fa fa-check"></i>Creating a Package in RStudio</a></li>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#using-the-package"><i class="fa fa-check"></i>Using the Package</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="sharing.html"><a href="sharing.html#r-shiny-applications"><i class="fa fa-check"></i><b>10.6</b> R Shiny Applications</a>
<ul>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#how-to-create-an-r-shiny-application-in-the-rstudio-ide"><i class="fa fa-check"></i>How to Create an R Shiny Application in the RStudio IDE</a></li>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#example-of-an-r-shiny-application-for-plotting-points"><i class="fa fa-check"></i>Example of an R Shiny Application for Plotting Points</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/jeffsocal/asms_r_intro" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Introduction to R for Mass Spectrometrists</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="mass-spectrometry-r-packages-work-in-progress" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">Chapter 9</span> Mass Spectrometry R Packages (work in progress)<a href="mass-spectrometry-r-packages-work-in-progress.html#mass-spectrometry-r-packages-work-in-progress" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<!--# HEATH -->
<div id="xcms" class="section level2 hasAnchor" number="9.1">
<h2><span class="header-section-number">9.1</span> xcms<a href="mass-spectrometry-r-packages-work-in-progress.html#xcms" class="anchor-section" aria-label="Anchor link to header"></a></h2>

</div>
<div id="deqms" class="section level2 hasAnchor" number="9.2">
<h2><span class="header-section-number">9.2</span> deqms<a href="mass-spectrometry-r-packages-work-in-progress.html#deqms" class="anchor-section" aria-label="Anchor link to header"></a></h2>
</div>
<div id="msempire" class="section level2 hasAnchor" number="9.3">
<h2><span class="header-section-number">9.3</span> msempire<a href="mass-spectrometry-r-packages-work-in-progress.html#msempire" class="anchor-section" aria-label="Anchor link to header"></a></h2>
</div>
<div id="msqrob" class="section level2 hasAnchor" number="9.4">
<h2><span class="header-section-number">9.4</span> msqrob<a href="mass-spectrometry-r-packages-work-in-progress.html#msqrob" class="anchor-section" aria-label="Anchor link to header"></a></h2>
</div>
<div id="msstatstmt" class="section level2 hasAnchor" number="9.5">
<h2><span class="header-section-number">9.5</span> MSstatsTMT<a href="mass-spectrometry-r-packages-work-in-progress.html#msstatstmt" class="anchor-section" aria-label="Anchor link to header"></a></h2>

</div>
<div id="msstats" class="section level2 hasAnchor" number="9.6">
<h2><span class="header-section-number">9.6</span> MSstats<a href="mass-spectrometry-r-packages-work-in-progress.html#msstats" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The MSstats package is an R package designed for the analysis of label-free mass spectrometry data. It provides a wide range of statistical tools for the analysis of protein abundance data, including normalization, missing value imputation, quality control, and differential expression analysis. MSstats provides a powerful and flexible way to analyze mass spectrometry data, making it an essential tool for researchers in the field.</p>
<table>
<colgroup>
<col width="22%" />
<col width="77%" />
</colgroup>
<tbody>
<tr class="odd">
<td><strong>Documentation</strong></td>
<td><a href="https://msstats.org">web</a>, <a href="https://bioconductor.org/packages/release/bioc/manuals/MSstats/man/MSstats.pdf">PDF</a></td>
</tr>
<tr class="even">
<td><strong>Code</strong></td>
<td><a href="https://www.bioconductor.org/packages/release/bioc/html/MSstats.html">Bioconductor</a></td>
</tr>
<tr class="odd">
<td><strong>Literature</strong></td>
<td><em>Bioinformatics</em> 30.17 (2014): 2524-2526</td>
</tr>
</tbody>
</table>
<div id="installation-1" class="section level3 hasAnchor" number="9.6.1">
<h3><span class="header-section-number">9.6.1</span> Installation<a href="mass-spectrometry-r-packages-work-in-progress.html#installation-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To install MSstats, you can use the following code:</p>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb137-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb137-2"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb137-2" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb137-3"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb137-3" tabindex="-1"></a></span>
<span id="cb137-4"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb137-4" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;MSstats&quot;</span>)</span></code></pre></div>
</div>
<div id="loading-data" class="section level3 hasAnchor" number="9.6.2">
<h3><span class="header-section-number">9.6.2</span> Loading Data<a href="mass-spectrometry-r-packages-work-in-progress.html#loading-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To load your data into MSstats, you can use the following code:</p>
<div class="sourceCode" id="cb138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb138-1"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb138-1" tabindex="-1"></a><span class="fu">library</span>(MSstats)</span>
<span id="cb138-2"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb138-2" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;QCMAPData&quot;</span>)</span>
<span id="cb138-3"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb138-3" tabindex="-1"></a>data <span class="ot">&lt;-</span> QCMAPData<span class="sc">$</span>data</span>
<span id="cb138-4"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb138-4" tabindex="-1"></a>design <span class="ot">&lt;-</span> QCMAPData<span class="sc">$</span>design</span>
<span id="cb138-5"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb138-5" tabindex="-1"></a><span class="fu">head</span>(data)</span></code></pre></div>
</div>
<div id="quality-control" class="section level3 hasAnchor" number="9.6.3">
<h3><span class="header-section-number">9.6.3</span> Quality Control<a href="mass-spectrometry-r-packages-work-in-progress.html#quality-control" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To perform quality control on your data, you can use the following code:</p>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb139-1"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb139-1" tabindex="-1"></a>QCmetrics <span class="ot">&lt;-</span> MSstats<span class="sc">:::</span><span class="fu">calculateQCmetrics</span>(data, <span class="at">featureLevel =</span> <span class="st">&quot;protein&quot;</span>)</span>
<span id="cb139-2"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb139-2" tabindex="-1"></a>MSstats<span class="sc">:::</span><span class="fu">plotQCmetrics</span>(QCmetrics, <span class="at">plotType =</span> <span class="st">&quot;boxplot&quot;</span>,</span>
<span id="cb139-3"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb139-3" tabindex="-1"></a>                        <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">200000</span>), <span class="at">ylab =</span> <span class="st">&quot;Total Ion Current&quot;</span>)</span></code></pre></div>
</div>
<div id="differential-expression-analysis" class="section level3 hasAnchor" number="9.6.4">
<h3><span class="header-section-number">9.6.4</span> Differential Expression Analysis<a href="mass-spectrometry-r-packages-work-in-progress.html#differential-expression-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To perform differential expression analysis on your data, you can use the following code:</p>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb140-1" tabindex="-1"></a>MSstatsResult <span class="ot">&lt;-</span> MSstats<span class="sc">::</span><span class="fu">MSstats</span>(data, design,</span>
<span id="cb140-2"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb140-2" tabindex="-1"></a>                                  <span class="at">normalizationMethod =</span> <span class="st">&quot;quantile&quot;</span>,</span>
<span id="cb140-3"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb140-3" tabindex="-1"></a>                                  <span class="at">normalization =</span> <span class="st">&quot;median&quot;</span>,</span>
<span id="cb140-4"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb140-4" tabindex="-1"></a>                                  <span class="at">msLevel =</span> <span class="dv">1</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb140-5"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb140-5" tabindex="-1"></a><span class="fu">summary</span>(MSstatsResult)</span></code></pre></div>
</div>
<div id="conclusion" class="section level3 hasAnchor" number="9.6.5">
<h3><span class="header-section-number">9.6.5</span> Conclusion<a href="mass-spectrometry-r-packages-work-in-progress.html#conclusion" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Overall, MSstats is a powerful and flexible R package for the analysis of label-free mass spectrometry data. It provides a wide range of statistical tools for the analysis of protein abundance data, making it an essential tool for researchers in the field. With its easy-to-use interface and comprehensive documentation, MSstats is a must-have tool for anyone working with mass spectrometry data.</p>

</div>
</div>
<div id="msgrob2" class="section level2 hasAnchor" number="9.7">
<h2><span class="header-section-number">9.7</span> msgrob2<a href="mass-spectrometry-r-packages-work-in-progress.html#msgrob2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><code>msgrob2</code> is an R package that provides functions to perform robust estimation in linear models with missing data. With the help of the Expectation-Maximization (EM) algorithm, the package estimates the parameters of the linear model and imputes the missing data. Additionally, the package offers robust methods for estimating the covariance matrix, including the Minimum Covariance Determinant (MCD) estimator and the S-estimator.</p>
<p>The <code>msgrob2</code> package is particularly useful in situations where data is missing from a linear model. The EM algorithm implemented in the package is a powerful tool for imputing missing data, and the robust covariance estimators allow for a better understanding of the data. The package is designed to provide efficient and accurate results when working with incomplete data, making it an essential tool for researchers and data analysts.</p>
<table>
<colgroup>
<col width="25%" />
<col width="75%" />
</colgroup>
<tbody>
<tr class="odd">
<td><strong>Documentation</strong></td>
<td><a href="https://www.bioconductor.org/packages/release/bioc/manuals/msqrob2/man/msqrob2.pdf">PDF</a></td>
</tr>
<tr class="even">
<td><strong>Code</strong></td>
<td><a href="https://www.bioconductor.org/packages/release/bioc/html/msqrob2.html">Bioconductor</a></td>
</tr>
<tr class="odd">
<td><strong>Literature</strong></td>
<td><em>Molecular &amp; Cellular Proteomics</em>, <strong>15</strong>(2), 657-668., <em>Molecular &amp; Cellular Proteomics</em>, <strong>19</strong>(7), 1209-1219., <em>Analytical Chemistry</em>, <strong>92</strong>(9), 6278–6287.</td>
</tr>
</tbody>
</table>
<div id="code-examples" class="section level3 hasAnchor" number="9.7.1">
<h3><span class="header-section-number">9.7.1</span> Code Examples<a href="mass-spectrometry-r-packages-work-in-progress.html#code-examples" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="installing-package" class="section level4 hasAnchor" number="9.7.1.1">
<h4><span class="header-section-number">9.7.1.1</span> Installing Package<a href="mass-spectrometry-r-packages-work-in-progress.html#installing-package" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>To install the <code>msgrob2</code> package, you can use the following code:</p>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb141-1" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;msgrob2&quot;</span>)</span></code></pre></div>
</div>
<div id="loading-package" class="section level4 hasAnchor" number="9.7.1.2">
<h4><span class="header-section-number">9.7.1.2</span> Loading Package<a href="mass-spectrometry-r-packages-work-in-progress.html#loading-package" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>After installation, you can load the <code>msgrob2</code> package using the following code:</p>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb142-1" tabindex="-1"></a><span class="fu">library</span>(msgrob2)</span></code></pre></div>
</div>
<div id="estimating-parameters" class="section level4 hasAnchor" number="9.7.1.3">
<h4><span class="header-section-number">9.7.1.3</span> Estimating Parameters<a href="mass-spectrometry-r-packages-work-in-progress.html#estimating-parameters" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>To estimate the parameters of a linear model with missing data using <code>msgrob2</code>, you can use the <code>msgrob()</code> function. The following code demonstrates how to use <code>msgrob()</code>:</p>
<div class="sourceCode" id="cb143"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb143-1"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb143-1" tabindex="-1"></a><span class="co"># Loading Example Data</span></span>
<span id="cb143-2"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb143-2" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;exMiss&quot;</span>)</span>
<span id="cb143-3"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb143-3" tabindex="-1"></a></span>
<span id="cb143-4"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb143-4" tabindex="-1"></a><span class="co"># Estimating Parameters</span></span>
<span id="cb143-5"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb143-5" tabindex="-1"></a>model <span class="ot">&lt;-</span> <span class="fu">lm</span>(y <span class="sc">~</span> x1 <span class="sc">+</span> x2 <span class="sc">+</span> x3, <span class="at">data =</span> exMiss)</span>
<span id="cb143-6"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb143-6" tabindex="-1"></a><span class="fu">msgrob</span>(model)</span></code></pre></div>
</div>
<div id="estimating-covariance-matrix" class="section level4 hasAnchor" number="9.7.1.4">
<h4><span class="header-section-number">9.7.1.4</span> Estimating Covariance Matrix<a href="mass-spectrometry-r-packages-work-in-progress.html#estimating-covariance-matrix" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>To estimate the covariance matrix using <code>msgrob2</code>, you can use the <code>covrob()</code> function. The following code demonstrates how to use <code>covrob()</code>:</p>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb144-1" tabindex="-1"></a><span class="co"># Loading Example Data</span></span>
<span id="cb144-2"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb144-2" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;exMiss&quot;</span>)</span>
<span id="cb144-3"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb144-3" tabindex="-1"></a></span>
<span id="cb144-4"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb144-4" tabindex="-1"></a><span class="co"># Estimating Covariance Matrix</span></span>
<span id="cb144-5"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb144-5" tabindex="-1"></a><span class="fu">covrob</span>(exMiss[,<span class="dv">2</span><span class="sc">:</span><span class="dv">4</span>], <span class="at">method =</span> <span class="st">&quot;MCD&quot;</span>)</span></code></pre></div>
<p>In summary, <code>msgrob2</code> is a comprehensive R package that provides an array of functions for robust estimation in linear models with missing data. The package’s implementation of the EM algorithm and robust covariance estimators make it an essential tool for researchers and data analysts working with incomplete data. The package is easy to install and use, with code examples readily available for reference.</p>

</div>
</div>
</div>
<div id="statspro-r-package" class="section level2 hasAnchor" number="9.8">
<h2><span class="header-section-number">9.8</span> StatsPro R Package<a href="mass-spectrometry-r-packages-work-in-progress.html#statspro-r-package" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<table>
<colgroup>
<col width="22%" />
<col width="77%" />
</colgroup>
<tbody>
<tr class="odd">
<td><strong>Documentation</strong></td>
<td><a href="https://github.com/YanglabWCH/StatsPro#Readme">github</a>, <a href="https://github.com/YanglabWCH/StatsPro/blob/main/StatsPro_Manual.pdf">PDF</a></td>
</tr>
<tr class="even">
<td><strong>Code</strong></td>
<td><a href="https://github.com/YanglabWCH/StatsPro">github</a></td>
</tr>
<tr class="odd">
<td><strong>Literature</strong></td>
<td><em>Journal of Proteomics</em> 250 (2022): 104386.</td>
</tr>
</tbody>
</table>

</div>
<div id="tidyproteomics" class="section level2 hasAnchor" number="9.9">
<h2><span class="header-section-number">9.9</span> Tidyproteomics<a href="mass-spectrometry-r-packages-work-in-progress.html#tidyproteomics" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The tidyproteomics R package is a tool that provides a set of functions to preprocess and analyze proteomics data using the tidy data framework. This package is built on top of the tidyverse and Bioconductor packages, which are widely used in the R community for data manipulation and analysis.</p>
<p>Some of the main features of the tidyproteomics package include:</p>
<ul>
<li>Data preprocessing functions for common tasks such as filtering, normalization, and imputation.</li>
<li>Functions for quality assessment and visualization of proteomics data.</li>
<li>Integration with other Bioconductor packages for downstream analysis such as differential expression analysis and pathway analysis.</li>
</ul>
<table>
<colgroup>
<col width="22%" />
<col width="77%" />
</colgroup>
<tbody>
<tr class="odd">
<td><strong>Documentation</strong></td>
<td><a href="https://jeffsocal.github.io/tidyproteomics/">github</a></td>
</tr>
<tr class="even">
<td><strong>Code</strong></td>
<td><a href="https://github.com/jeffsocal/tidyproteomics">github</a></td>
</tr>
<tr class="odd">
<td><strong>Literature</strong></td>
<td></td>
</tr>
</tbody>
</table>
<div id="loading-data-1" class="section level3 hasAnchor" number="9.9.1">
<h3><span class="header-section-number">9.9.1</span> Loading Data<a href="mass-spectrometry-r-packages-work-in-progress.html#loading-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To load your data into tidyproteomics, you can use the following code:</p>
<div class="sourceCode" id="cb145"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb145-1"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb145-1" tabindex="-1"></a><span class="co"># Load the tidyproteomics package</span></span>
<span id="cb145-2"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb145-2" tabindex="-1"></a><span class="fu">library</span>(tidyproteomics)</span>
<span id="cb145-3"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb145-3" tabindex="-1"></a></span>
<span id="cb145-4"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb145-4" tabindex="-1"></a><span class="co"># Import data</span></span>
<span id="cb145-5"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb145-5" tabindex="-1"></a>data_proteins <span class="ot">&lt;-</span> <span class="st">&quot;path_to_data.xlsx&quot;</span> <span class="sc">%&gt;%</span></span>
<span id="cb145-6"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb145-6" tabindex="-1"></a>   <span class="fu">import</span>(<span class="st">&quot;ProteomeDiscoverer&quot;</span>, <span class="st">&quot;proteins&quot;</span>) </span></code></pre></div>
</div>
<div id="data-summaries" class="section level3 hasAnchor" number="9.9.2">
<h3><span class="header-section-number">9.9.2</span> Data Summaries<a href="mass-spectrometry-r-packages-work-in-progress.html#data-summaries" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb146-1" tabindex="-1"></a>rdata <span class="ot">&lt;-</span> rdata <span class="sc">%&gt;%</span></span>
<span id="cb146-2"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb146-2" tabindex="-1"></a>  <span class="co"># plot some simple summary stats</span></span>
<span id="cb146-3"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb146-3" tabindex="-1"></a>  <span class="fu">plot_counts</span>(<span class="at">destination =</span> <span class="st">&quot;png&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb146-4"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb146-4" tabindex="-1"></a>  <span class="fu">plot_quantrank</span>(<span class="at">destination =</span> <span class="st">&quot;png&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb146-5"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb146-5" tabindex="-1"></a>  <span class="fu">plot_venn</span>(<span class="at">destination =</span> <span class="st">&quot;png&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb146-6"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb146-6" tabindex="-1"></a>  <span class="fu">plot_euler</span>(<span class="at">destination =</span> <span class="st">&quot;png&quot;</span>)</span></code></pre></div>
<p>Summary Stats</p>
<div class="sourceCode" id="cb147"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb147-1"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb147-1" tabindex="-1"></a>rdata <span class="ot">&lt;-</span> rdata <span class="sc">%&gt;%</span></span>
<span id="cb147-2"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb147-2" tabindex="-1"></a>  <span class="co"># save a table of simple summary stats</span></span>
<span id="cb147-3"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb147-3" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="st">&quot;sample&quot;</span>, <span class="at">destination =</span> <span class="st">&quot;save&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb147-4"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb147-4" tabindex="-1"></a>  <span class="co"># save a report on contamination</span></span>
<span id="cb147-5"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb147-5" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">contamination =</span> <span class="st">&quot;CRAP&quot;</span>, <span class="at">destination =</span> <span class="st">&quot;save&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb147-6"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb147-6" tabindex="-1"></a>  <span class="co"># remove contamination</span></span>
<span id="cb147-7"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb147-7" tabindex="-1"></a>  <span class="fu">subset</span>(<span class="sc">!</span>description <span class="sc">%like%</span> <span class="st">&quot;^CRAP&quot;</span>)</span></code></pre></div>
</div>
<div id="normalization-and-imputation" class="section level3 hasAnchor" number="9.9.3">
<h3><span class="header-section-number">9.9.3</span> Normalization and Imputation<a href="mass-spectrometry-r-packages-work-in-progress.html#normalization-and-imputation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb148"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb148-1"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb148-1" tabindex="-1"></a>rdata <span class="ot">&lt;-</span> rdata <span class="sc">%&gt;%</span></span>
<span id="cb148-2"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb148-2" tabindex="-1"></a>  <span class="co"># normalize via several methods, best method will be automatically selected</span></span>
<span id="cb148-3"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb148-3" tabindex="-1"></a>  <span class="fu">normalize</span>(<span class="fu">c</span>(<span class="st">&quot;median&quot;</span>,<span class="st">&quot;linear&quot;</span>,<span class="st">&quot;limma&quot;</span>,<span class="st">&quot;randomforest&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb148-4"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb148-4" tabindex="-1"></a>  <span class="co"># impute with a minimum value (this is a knock-out)</span></span>
<span id="cb148-5"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb148-5" tabindex="-1"></a>  <span class="fu">impute</span>(base<span class="sc">::</span>min)</span>
<span id="cb148-6"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb148-6" tabindex="-1"></a>  <span class="co"># plot visualizations comparing normalization methods</span></span>
<span id="cb148-7"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb148-7" tabindex="-1"></a>  <span class="fu">plot_normalization</span>(<span class="at">destination =</span> <span class="st">&quot;png&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb148-8"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb148-8" tabindex="-1"></a>  <span class="fu">plot_variation_cv</span>(<span class="at">destination =</span> <span class="st">&quot;png&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb148-9"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb148-9" tabindex="-1"></a>  <span class="fu">plot_variation_pca</span>(<span class="at">destination =</span> <span class="st">&quot;png&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb148-10"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb148-10" tabindex="-1"></a>  <span class="fu">plot_dynamic_range</span>(<span class="at">destination =</span> <span class="st">&quot;png&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb148-11"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb148-11" tabindex="-1"></a>  <span class="co"># plot visualizations of unbiased clustering</span></span>
<span id="cb148-12"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb148-12" tabindex="-1"></a>  <span class="fu">plot_heatmap</span>(<span class="at">destination =</span> <span class="st">&quot;png&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb148-13"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb148-13" tabindex="-1"></a>  <span class="fu">plot_pca</span>(<span class="at">destination =</span> <span class="st">&quot;png&quot;</span>)</span></code></pre></div>
</div>
<div id="expression-analysis" class="section level3 hasAnchor" number="9.9.4">
<h3><span class="header-section-number">9.9.4</span> Expression Analysis<a href="mass-spectrometry-r-packages-work-in-progress.html#expression-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb149"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb149-1"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb149-1" tabindex="-1"></a>rdata <span class="ot">&lt;-</span> rdata <span class="sc">%&gt;%</span></span>
<span id="cb149-2"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb149-2" tabindex="-1"></a>  <span class="co"># calculate the expression between experiment: ko and control: wt</span></span>
<span id="cb149-3"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb149-3" tabindex="-1"></a>  <span class="fu">expression</span>(kndw<span class="sc">/</span>ctrl) <span class="sc">%&gt;%</span></span>
<span id="cb149-4"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb149-4" tabindex="-1"></a>  <span class="co"># plot the expression analysis</span></span>
<span id="cb149-5"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb149-5" tabindex="-1"></a>  <span class="fu">plot_volcano</span>(kndw<span class="sc">/</span>ctrl, <span class="at">destination =</span> <span class="st">&quot;png&quot;</span>, <span class="at">significance_column =</span> <span class="st">&quot;p_value&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb149-6"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb149-6" tabindex="-1"></a>  <span class="fu">plot_proportion</span>(kndw<span class="sc">/</span>ctrl, <span class="at">destination =</span> <span class="st">&quot;png&quot;</span>)</span></code></pre></div>
<p>Enrichment Analysis</p>
<div class="sourceCode" id="cb150"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb150-1"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb150-1" tabindex="-1"></a>rdata <span class="ot">&lt;-</span> rdata <span class="sc">%&gt;%</span></span>
<span id="cb150-2"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb150-2" tabindex="-1"></a>  <span class="co"># calculate the enrichment of the GO term(s) using the results</span></span>
<span id="cb150-3"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb150-3" tabindex="-1"></a>  <span class="co"># from the expression analysis</span></span>
<span id="cb150-4"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb150-4" tabindex="-1"></a>  <span class="fu">enrichment</span>(kndw<span class="sc">/</span>ctrl, <span class="at">.term =</span> <span class="st">&quot;biological_process&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb150-5"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb150-5" tabindex="-1"></a>  <span class="fu">enrichment</span>(kndw<span class="sc">/</span>ctrl, <span class="at">.term =</span> <span class="st">&quot;cellular_component&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb150-6"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb150-6" tabindex="-1"></a>  <span class="fu">enrichment</span>(kndw<span class="sc">/</span>ctrl, <span class="at">.term =</span> <span class="st">&quot;molecular_function&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb150-7"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb150-7" tabindex="-1"></a>  <span class="co"># plot the enrichment analysis</span></span>
<span id="cb150-8"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb150-8" tabindex="-1"></a>  <span class="fu">plot_enrichment</span>(kndw<span class="sc">/</span>ctrl, <span class="at">.term =</span> <span class="st">&quot;biological_process&quot;</span>, <span class="at">destination =</span> <span class="st">&quot;png&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb150-9"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb150-9" tabindex="-1"></a>  <span class="fu">plot_enrichment</span>(kndw<span class="sc">/</span>ctrl, <span class="at">.term =</span> <span class="st">&quot;cellular_component&quot;</span>, <span class="at">destination =</span> <span class="st">&quot;png&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb150-10"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb150-10" tabindex="-1"></a>  <span class="fu">plot_enrichment</span>(kndw<span class="sc">/</span>ctrl, <span class="at">.term =</span> <span class="st">&quot;molecular_function&quot;</span>, <span class="at">destination =</span> <span class="st">&quot;png&quot;</span>) </span></code></pre></div>
</div>
<div id="advanced" class="section level3 hasAnchor" number="9.9.5">
<h3><span class="header-section-number">9.9.5</span> Advanced<a href="mass-spectrometry-r-packages-work-in-progress.html#advanced" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="plot-quantitation-rank-with-log2-cutoff" class="section level4 hasAnchor" number="9.9.5.1">
<h4><span class="header-section-number">9.9.5.1</span> Plot Quantitation-Rank with Log2 Cutoff<a href="mass-spectrometry-r-packages-work-in-progress.html#plot-quantitation-rank-with-log2-cutoff" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb151"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb151-1"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb151-1" tabindex="-1"></a><span class="co"># SUPPLEMENTAL FIGURES</span></span>
<span id="cb151-2"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb151-2" tabindex="-1"></a><span class="co"># plot an alternate quantitative ranking</span></span>
<span id="cb151-3"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb151-3" tabindex="-1"></a>rdata <span class="sc">%&gt;%</span></span>
<span id="cb151-4"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb151-4" tabindex="-1"></a>  <span class="fu">plot_quantrank</span>(<span class="at">display_filter =</span> <span class="st">&quot;log2_foldchange&quot;</span>,</span>
<span id="cb151-5"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb151-5" tabindex="-1"></a>                 <span class="at">display_cutoff =</span> <span class="dv">5</span>)</span>
<span id="cb151-6"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb151-6" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;plot_proteins_quantitation_rank_filtered.png&quot;</span>,</span>
<span id="cb151-7"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb151-7" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">6</span>, <span class="at">h =</span> <span class="dv">4</span>)</span></code></pre></div>
</div>
<div id="plot-comparison-of-two-expressions" class="section level4 hasAnchor" number="9.9.5.2">
<h4><span class="header-section-number">9.9.5.2</span> Plot Comparison of Two Expressions<a href="mass-spectrometry-r-packages-work-in-progress.html#plot-comparison-of-two-expressions" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb152"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb152-1"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb152-1" tabindex="-1"></a><span class="co"># import the data again to now avoid imputation</span></span>
<span id="cb152-2"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb152-2" tabindex="-1"></a>rdata <span class="ot">&lt;-</span> <span class="fu">path_to_package_data</span>(<span class="st">&quot;p97KD_HCT116&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb152-3"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb152-3" tabindex="-1"></a>  <span class="co"># import the data set</span></span>
<span id="cb152-4"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb152-4" tabindex="-1"></a>  <span class="fu">import</span>(<span class="st">&#39;ProteomeDiscoverer&#39;</span>, <span class="st">&#39;proteins&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb152-5"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb152-5" tabindex="-1"></a>  <span class="co"># change the labels on the samples containing &#39;ko&#39;</span></span>
<span id="cb152-6"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb152-6" tabindex="-1"></a>  <span class="fu">reassign</span>(<span class="st">&quot;sample&quot;</span>, <span class="st">&quot;ctl&quot;</span>, <span class="st">&quot;ctrl&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb152-7"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb152-7" tabindex="-1"></a>  <span class="fu">reassign</span>(<span class="st">&quot;sample&quot;</span>, <span class="st">&quot;p97&quot;</span>, <span class="st">&quot;kndw&quot;</span>)</span>
<span id="cb152-8"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb152-8" tabindex="-1"></a></span>
<span id="cb152-9"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb152-9" tabindex="-1"></a><span class="co"># run a an expression analysis using a t.test statistical comparison</span></span>
<span id="cb152-10"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb152-10" tabindex="-1"></a>tbl_expression_ttest <span class="ot">&lt;-</span> rdata <span class="sc">%&gt;%</span></span>
<span id="cb152-11"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb152-11" tabindex="-1"></a>  <span class="fu">expression</span>(kndw<span class="sc">/</span>ctrl, <span class="at">.method =</span> stats<span class="sc">::</span>t.test) <span class="sc">%&gt;%</span></span>
<span id="cb152-12"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb152-12" tabindex="-1"></a>  <span class="co"># export the results table to the assigned object</span></span>
<span id="cb152-13"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb152-13" tabindex="-1"></a>  <span class="fu">export_analysis</span>(kndw<span class="sc">/</span>ctrl, <span class="at">.analysis =</span> <span class="st">&quot;expression&quot;</span>)</span>
<span id="cb152-14"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb152-14" tabindex="-1"></a></span>
<span id="cb152-15"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb152-15" tabindex="-1"></a><span class="co"># run a an expression analysis using the limma statistical method</span></span>
<span id="cb152-16"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb152-16" tabindex="-1"></a>tbl_expression_limma <span class="ot">&lt;-</span> rdata <span class="sc">%&gt;%</span></span>
<span id="cb152-17"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb152-17" tabindex="-1"></a>  <span class="fu">expression</span>(kndw<span class="sc">/</span>ctrl, <span class="at">.method =</span> <span class="st">&quot;limma&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb152-18"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb152-18" tabindex="-1"></a>  <span class="co"># export the results table to the assigned object</span></span>
<span id="cb152-19"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb152-19" tabindex="-1"></a>  <span class="fu">export_analysis</span>(kndw<span class="sc">/</span>ctrl, <span class="at">.analysis =</span> <span class="st">&quot;expression&quot;</span>)</span>
<span id="cb152-20"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb152-20" tabindex="-1"></a></span>
<span id="cb152-21"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb152-21" tabindex="-1"></a><span class="co"># plot the two expression tables two compare similarities between methods</span></span>
<span id="cb152-22"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb152-22" tabindex="-1"></a><span class="fu">plot_compexp</span>(tbl_expression_ttest,</span>
<span id="cb152-23"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb152-23" tabindex="-1"></a>             tbl_expression_limma,</span>
<span id="cb152-24"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb152-24" tabindex="-1"></a>             <span class="at">labels_column =</span> <span class="st">&quot;gene_name&quot;</span>,</span>
<span id="cb152-25"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb152-25" tabindex="-1"></a>             <span class="at">log2fc_min =</span> <span class="dv">1</span>, <span class="at">significance_column =</span> <span class="st">&quot;p_value&quot;</span>) <span class="sc">+</span></span>
<span id="cb152-26"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb152-26" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;(log2 FC) Wilcoxon Rank Sum&quot;</span>,</span>
<span id="cb152-27"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb152-27" tabindex="-1"></a>                <span class="at">y =</span> <span class="st">&quot;(log2 FC) Emperical Bayes (limma)&quot;</span>,</span>
<span id="cb152-28"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb152-28" tabindex="-1"></a>                <span class="at">title =</span> <span class="st">&quot;Hela p97 KD ~ Ctrl&quot;</span>)</span>
<span id="cb152-29"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb152-29" tabindex="-1"></a></span>
<span id="cb152-30"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb152-30" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;plot_enrichment_comparison.png&quot;</span>,</span>
<span id="cb152-31"><a href="mass-spectrometry-r-packages-work-in-progress.html#cb152-31" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">6</span>, <span class="at">h =</span> <span class="dv">4</span>)</span></code></pre></div>
<p>Overall, the tidyproteomics package provides a useful set of tools for preprocessing and analyzing proteomics data using the tidy data framework in R.</p>

</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="mass-spectrometry-data.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="sharing.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/jeffsocal/asms_r_intro/edit/BRANCH/08_01_ms_packages.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
