<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 10 Mass Spectrometry | Learning R for Mass Spectrometry</title>
  <meta name="description" content="An introductory approach to using R in the analysis of mass spectrometry related data." />
  <meta name="generator" content="bookdown 0.33 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 10 Mass Spectrometry | Learning R for Mass Spectrometry" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="An introductory approach to using R in the analysis of mass spectrometry related data." />
  <meta name="github-repo" content="jeffsocal/ASMS_R_Basics" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 10 Mass Spectrometry | Learning R for Mass Spectrometry" />
  
  <meta name="twitter:description" content="An introductory approach to using R in the analysis of mass spectrometry related data." />
  

<meta name="author" content="Jeff Jones, Heath Patterson, Ryan Benz" />


<meta name="date" content="2023-06-02" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="sharing.html"/>
<link rel="next" href="answers-to-exercises.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="libs/plotly-binding-4.10.1/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-2.11.1/plotly-latest.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Learning R for Mass Spec.</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="" data-path="acknowledgements.html"><a href="acknowledgements.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="" data-path="contents.html"><a href="contents.html"><i class="fa fa-check"></i>Contents</a></li>
<li class="chapter" data-level="1" data-path="introduction-to-r.html"><a href="introduction-to-r.html"><i class="fa fa-check"></i><b>1</b> Introduction to R</a>
<ul>
<li class="chapter" data-level="" data-path="introduction-to-r.html"><a href="introduction-to-r.html#why-choose-it"><i class="fa fa-check"></i>Why choose it?</a></li>
<li class="chapter" data-level="" data-path="introduction-to-r.html"><a href="introduction-to-r.html#what-you-can-do-with-it"><i class="fa fa-check"></i>What you can do with it?</a></li>
<li class="chapter" data-level="" data-path="introduction-to-r.html"><a href="introduction-to-r.html#the-r-learning-curve"><i class="fa fa-check"></i>The R Learning Curve</a></li>
<li class="chapter" data-level="" data-path="introduction-to-r.html"><a href="introduction-to-r.html#learning-to-code"><i class="fa fa-check"></i>Learning to code</a></li>
<li class="chapter" data-level="1.1" data-path="introduction-to-r.html"><a href="introduction-to-r.html#alternatives"><i class="fa fa-check"></i><b>1.1</b> Alternatives</a></li>
<li class="chapter" data-level="1.2" data-path="introduction-to-r.html"><a href="introduction-to-r.html#resources"><i class="fa fa-check"></i><b>1.2</b> Resources</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="introduction-to-r.html"><a href="introduction-to-r.html#online"><i class="fa fa-check"></i><b>1.2.1</b> Online</a></li>
<li class="chapter" data-level="1.2.2" data-path="introduction-to-r.html"><a href="introduction-to-r.html#in-print"><i class="fa fa-check"></i><b>1.2.2</b> In Print</a></li>
<li class="chapter" data-level="1.2.3" data-path="introduction-to-r.html"><a href="introduction-to-r.html#organizations"><i class="fa fa-check"></i><b>1.2.3</b> Organizations</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="installation.html"><a href="installation.html"><i class="fa fa-check"></i><b>2</b> Installation</a>
<ul>
<li class="chapter" data-level="2.1" data-path="installation.html"><a href="installation.html#r-interpreter"><i class="fa fa-check"></i><b>2.1</b> R interpreter</a>
<ul>
<li class="chapter" data-level="" data-path="installation.html"><a href="installation.html#steps"><i class="fa fa-check"></i>Steps</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="installation.html"><a href="installation.html#r-studio-ide"><i class="fa fa-check"></i><b>2.2</b> R Studio IDE</a>
<ul>
<li class="chapter" data-level="" data-path="installation.html"><a href="installation.html#steps-1"><i class="fa fa-check"></i>Steps</a></li>
<li class="chapter" data-level="" data-path="installation.html"><a href="installation.html#ide-layout"><i class="fa fa-check"></i>IDE Layout</a></li>
<li class="chapter" data-level="" data-path="installation.html"><a href="installation.html#usage"><i class="fa fa-check"></i>Usage</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="installation.html"><a href="installation.html#r-packages"><i class="fa fa-check"></i><b>2.3</b> R Packages</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="installation.html"><a href="installation.html#exploring"><i class="fa fa-check"></i><b>2.3.1</b> Exploring</a></li>
<li class="chapter" data-level="2.3.2" data-path="installation.html"><a href="installation.html#installing"><i class="fa fa-check"></i><b>2.3.2</b> Installing</a></li>
<li class="chapter" data-level="" data-path="installation.html"><a href="installation.html#gotchas"><i class="fa fa-check"></i>Gotchas</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="installation.html"><a href="installation.html#for-this-book"><i class="fa fa-check"></i><b>2.4</b> For this Book</a>
<ul>
<li class="chapter" data-level="" data-path="installation.html"><a href="installation.html#tidyverse"><i class="fa fa-check"></i>tidyverse</a></li>
<li class="chapter" data-level="" data-path="installation.html"><a href="installation.html#ms-packages"><i class="fa fa-check"></i>MS Packages</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="r-studio-projects.html"><a href="r-studio-projects.html"><i class="fa fa-check"></i><b>3</b> R Studio Projects</a>
<ul>
<li class="chapter" data-level="3.1" data-path="r-studio-projects.html"><a href="r-studio-projects.html#creating"><i class="fa fa-check"></i><b>3.1</b> Creating</a></li>
<li class="chapter" data-level="3.2" data-path="r-studio-projects.html"><a href="r-studio-projects.html#organizing"><i class="fa fa-check"></i><b>3.2</b> Organizing</a>
<ul>
<li class="chapter" data-level="" data-path="r-studio-projects.html"><a href="r-studio-projects.html#data"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="r-studio-projects.html"><a href="r-studio-projects.html#scripts"><i class="fa fa-check"></i>Scripts</a></li>
<li class="chapter" data-level="" data-path="r-studio-projects.html"><a href="r-studio-projects.html#results"><i class="fa fa-check"></i>Results</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-studio-projects.html"><a href="r-studio-projects.html#exercises"><i class="fa fa-check"></i>Exercises</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="r-syntax.html"><a href="r-syntax.html"><i class="fa fa-check"></i><b>4</b> R Syntax</a>
<ul>
<li class="chapter" data-level="4.1" data-path="r-syntax.html"><a href="r-syntax.html#reserved-words"><i class="fa fa-check"></i><b>4.1</b> Reserved Words</a></li>
<li class="chapter" data-level="4.2" data-path="r-syntax.html"><a href="r-syntax.html#types"><i class="fa fa-check"></i><b>4.2</b> Types</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="r-syntax.html"><a href="r-syntax.html#comments"><i class="fa fa-check"></i><b>4.2.1</b> Comments</a></li>
<li class="chapter" data-level="4.2.2" data-path="r-syntax.html"><a href="r-syntax.html#strings"><i class="fa fa-check"></i><b>4.2.2</b> Strings</a></li>
<li class="chapter" data-level="4.2.3" data-path="r-syntax.html"><a href="r-syntax.html#numbers"><i class="fa fa-check"></i><b>4.2.3</b> Numbers</a></li>
<li class="chapter" data-level="4.2.4" data-path="r-syntax.html"><a href="r-syntax.html#operators"><i class="fa fa-check"></i><b>4.2.4</b> Operators</a></li>
<li class="chapter" data-level="4.2.5" data-path="r-syntax.html"><a href="r-syntax.html#variables"><i class="fa fa-check"></i><b>4.2.5</b> Variables</a></li>
<li class="chapter" data-level="4.2.6" data-path="r-syntax.html"><a href="r-syntax.html#statements"><i class="fa fa-check"></i><b>4.2.6</b> Statements</a></li>
<li class="chapter" data-level="4.2.7" data-path="r-syntax.html"><a href="r-syntax.html#functions"><i class="fa fa-check"></i><b>4.2.7</b> Functions</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="r-syntax.html"><a href="r-syntax.html#flow-control"><i class="fa fa-check"></i><b>4.3</b> Flow-Control</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="r-syntax.html"><a href="r-syntax.html#for-loop"><i class="fa fa-check"></i><b>4.3.1</b> For Loop</a></li>
<li class="chapter" data-level="4.3.2" data-path="r-syntax.html"><a href="r-syntax.html#if-else"><i class="fa fa-check"></i><b>4.3.2</b> If-Else</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-syntax.html"><a href="r-syntax.html#exercises-1"><i class="fa fa-check"></i>Exercises</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="r-objects.html"><a href="r-objects.html"><i class="fa fa-check"></i><b>5</b> R Objects</a>
<ul>
<li class="chapter" data-level="5.1" data-path="r-objects.html"><a href="r-objects.html#variable"><i class="fa fa-check"></i><b>5.1</b> Variable</a></li>
<li class="chapter" data-level="5.2" data-path="r-objects.html"><a href="r-objects.html#vector"><i class="fa fa-check"></i><b>5.2</b> Vector</a></li>
<li class="chapter" data-level="5.3" data-path="r-objects.html"><a href="r-objects.html#list"><i class="fa fa-check"></i><b>5.3</b> List</a></li>
<li class="chapter" data-level="5.4" data-path="r-objects.html"><a href="r-objects.html#matrix"><i class="fa fa-check"></i><b>5.4</b> Matrix</a></li>
<li class="chapter" data-level="5.5" data-path="r-objects.html"><a href="r-objects.html#data-frame"><i class="fa fa-check"></i><b>5.5</b> Data Frame</a></li>
<li class="chapter" data-level="5.6" data-path="r-objects.html"><a href="r-objects.html#data-table"><i class="fa fa-check"></i><b>5.6</b> Data Table</a></li>
<li class="chapter" data-level="5.7" data-path="r-objects.html"><a href="r-objects.html#tibbles"><i class="fa fa-check"></i><b>5.7</b> Tibbles</a></li>
<li class="chapter" data-level="" data-path="r-objects.html"><a href="r-objects.html#examine-the-contents"><i class="fa fa-check"></i>Examine the Contents</a></li>
<li class="chapter" data-level="" data-path="r-objects.html"><a href="r-objects.html#converting-objects"><i class="fa fa-check"></i>Converting Objects</a></li>
<li class="chapter" data-level="" data-path="r-objects.html"><a href="r-objects.html#exercises-2"><i class="fa fa-check"></i>Exercises</a></li>
<li class="chapter" data-level="" data-path="r-objects.html"><a href="r-objects.html#more-exercises"><i class="fa fa-check"></i>More Exercises</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="tidyverse-1.html"><a href="tidyverse-1.html"><i class="fa fa-check"></i><b>6</b> Tidyverse</a>
<ul>
<li class="chapter" data-level="6.1" data-path="tidyverse-1.html"><a href="tidyverse-1.html#core-packages"><i class="fa fa-check"></i><b>6.1</b> Core Packages</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="tidyverse-1.html"><a href="tidyverse-1.html#magrittr"><i class="fa fa-check"></i><b>6.1.1</b> magrittr</a></li>
<li class="chapter" data-level="6.1.2" data-path="tidyverse-1.html"><a href="tidyverse-1.html#tibble"><i class="fa fa-check"></i><b>6.1.2</b> tibble</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="tidyverse-1.html"><a href="tidyverse-1.html#importing"><i class="fa fa-check"></i><b>6.2</b> Importing</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="tidyverse-1.html"><a href="tidyverse-1.html#readr"><i class="fa fa-check"></i><b>6.2.1</b> readr</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="tidyverse-1.html"><a href="tidyverse-1.html#wrangling"><i class="fa fa-check"></i><b>6.3</b> Wrangling</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="tidyverse-1.html"><a href="tidyverse-1.html#dplyr"><i class="fa fa-check"></i><b>6.3.1</b> dplyr</a></li>
<li class="chapter" data-level="6.3.2" data-path="tidyverse-1.html"><a href="tidyverse-1.html#tidyr"><i class="fa fa-check"></i><b>6.3.2</b> tidyr</a></li>
<li class="chapter" data-level="6.3.3" data-path="tidyverse-1.html"><a href="tidyverse-1.html#purrr"><i class="fa fa-check"></i><b>6.3.3</b> purrr</a></li>
<li class="chapter" data-level="6.3.4" data-path="tidyverse-1.html"><a href="tidyverse-1.html#glue"><i class="fa fa-check"></i><b>6.3.4</b> glue</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="tidyverse-1.html"><a href="tidyverse-1.html#data-types"><i class="fa fa-check"></i><b>6.4</b> Data Types</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="tidyverse-1.html"><a href="tidyverse-1.html#stringr"><i class="fa fa-check"></i><b>6.4.1</b> stringr</a></li>
<li class="chapter" data-level="6.4.2" data-path="tidyverse-1.html"><a href="tidyverse-1.html#lubridate"><i class="fa fa-check"></i><b>6.4.2</b> lubridate</a></li>
<li class="chapter" data-level="6.4.3" data-path="tidyverse-1.html"><a href="tidyverse-1.html#forcats"><i class="fa fa-check"></i><b>6.4.3</b> forcats</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="tidyverse-1.html"><a href="tidyverse-1.html#summarizing"><i class="fa fa-check"></i><b>6.5</b> Summarizing</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="tidyverse-1.html"><a href="tidyverse-1.html#dplyr-1"><i class="fa fa-check"></i><b>6.5.1</b> dplyr</a></li>
<li class="chapter" data-level="6.5.2" data-path="tidyverse-1.html"><a href="tidyverse-1.html#ggplot2"><i class="fa fa-check"></i><b>6.5.2</b> ggplot2</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tidyverse-1.html"><a href="tidyverse-1.html#exercises-3"><i class="fa fa-check"></i>Exercises</a></li>
<li class="chapter" data-level="" data-path="tidyverse-1.html"><a href="tidyverse-1.html#more-exercises-1"><i class="fa fa-check"></i>More Exercises</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="data-wrangling.html"><a href="data-wrangling.html"><i class="fa fa-check"></i><b>7</b> Data Wrangling</a>
<ul>
<li class="chapter" data-level="7.1" data-path="data-wrangling.html"><a href="data-wrangling.html#tidy-data"><i class="fa fa-check"></i><b>7.1</b> Tidy Data</a></li>
<li class="chapter" data-level="7.2" data-path="data-wrangling.html"><a href="data-wrangling.html#example-walkthrough"><i class="fa fa-check"></i><b>7.2</b> Example Walkthrough</a>
<ul>
<li class="chapter" data-level="" data-path="data-wrangling.html"><a href="data-wrangling.html#read_"><i class="fa fa-check"></i>read_</a></li>
<li class="chapter" data-level="" data-path="data-wrangling.html"><a href="data-wrangling.html#separate"><i class="fa fa-check"></i>separate</a></li>
<li class="chapter" data-level="" data-path="data-wrangling.html"><a href="data-wrangling.html#mutate"><i class="fa fa-check"></i>mutate</a></li>
<li class="chapter" data-level="" data-path="data-wrangling.html"><a href="data-wrangling.html#pivot_"><i class="fa fa-check"></i>pivot_</a></li>
<li class="chapter" data-level="" data-path="data-wrangling.html"><a href="data-wrangling.html#write_"><i class="fa fa-check"></i>write_</a></li>
<li class="chapter" data-level="" data-path="data-wrangling.html"><a href="data-wrangling.html#non-pipe"><i class="fa fa-check"></i>Non-pipe</a></li>
<li class="chapter" data-level="" data-path="data-wrangling.html"><a href="data-wrangling.html#summarize"><i class="fa fa-check"></i>summarize</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-wrangling.html"><a href="data-wrangling.html#exercises-4"><i class="fa fa-check"></i>Exercises</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="data-visualization.html"><a href="data-visualization.html"><i class="fa fa-check"></i><b>8</b> Data Visualization</a>
<ul>
<li class="chapter" data-level="8.1" data-path="data-visualization.html"><a href="data-visualization.html#base-plot"><i class="fa fa-check"></i><b>8.1</b> Base: plot</a></li>
<li class="chapter" data-level="8.2" data-path="data-visualization.html"><a href="data-visualization.html#ggplot2-1"><i class="fa fa-check"></i><b>8.2</b> GGPlot2</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="data-visualization.html"><a href="data-visualization.html#syntax"><i class="fa fa-check"></i><b>8.2.1</b> Syntax</a></li>
<li class="chapter" data-level="8.2.2" data-path="data-visualization.html"><a href="data-visualization.html#basic-data-plotting"><i class="fa fa-check"></i><b>8.2.2</b> Basic Data Plotting</a></li>
<li class="chapter" data-level="8.2.3" data-path="data-visualization.html"><a href="data-visualization.html#data-distributions"><i class="fa fa-check"></i><b>8.2.3</b> Data Distributions</a></li>
<li class="chapter" data-level="8.2.4" data-path="data-visualization.html"><a href="data-visualization.html#extended-syntax"><i class="fa fa-check"></i><b>8.2.4</b> Extended Syntax</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="data-visualization.html"><a href="data-visualization.html#alternatives-1"><i class="fa fa-check"></i><b>8.3</b> Alternatives</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="data-visualization.html"><a href="data-visualization.html#lattice"><i class="fa fa-check"></i><b>8.3.1</b> lattice</a></li>
<li class="chapter" data-level="8.3.2" data-path="data-visualization.html"><a href="data-visualization.html#plotly"><i class="fa fa-check"></i><b>8.3.2</b> plotly</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-visualization.html"><a href="data-visualization.html#exercises-5"><i class="fa fa-check"></i>Exercises</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="sharing.html"><a href="sharing.html"><i class="fa fa-check"></i><b>9</b> Sharing</a>
<ul>
<li class="chapter" data-level="9.1" data-path="sharing.html"><a href="sharing.html#notebooks"><i class="fa fa-check"></i><b>9.1</b> Notebooks</a>
<ul>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#creating-1"><i class="fa fa-check"></i>Creating</a></li>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#sharing-1"><i class="fa fa-check"></i>Sharing</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="sharing.html"><a href="sharing.html#packrat"><i class="fa fa-check"></i><b>9.2</b> Packrat</a>
<ul>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#initiating"><i class="fa fa-check"></i>Initiating</a></li>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#installing-1"><i class="fa fa-check"></i>Installing</a></li>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#loading"><i class="fa fa-check"></i>Loading</a></li>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#updating"><i class="fa fa-check"></i>Updating</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="sharing.html"><a href="sharing.html#github-1"><i class="fa fa-check"></i><b>9.3</b> GitHub</a>
<ul>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#sharing-2"><i class="fa fa-check"></i>Sharing</a></li>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#sharing-3"><i class="fa fa-check"></i>Sharing</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="sharing.html"><a href="sharing.html#docker"><i class="fa fa-check"></i><b>9.4</b> Docker</a></li>
<li class="chapter" data-level="9.5" data-path="sharing.html"><a href="sharing.html#r-packages-1"><i class="fa fa-check"></i><b>9.5</b> R Packages</a>
<ul>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#creating-2"><i class="fa fa-check"></i>Creating</a></li>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#using"><i class="fa fa-check"></i>Using</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="sharing.html"><a href="sharing.html#r-shiny-applications"><i class="fa fa-check"></i><b>9.6</b> R Shiny Applications</a>
<ul>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#creating-3"><i class="fa fa-check"></i>Creating</a></li>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#example"><i class="fa fa-check"></i>Example</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html"><i class="fa fa-check"></i><b>10</b> Mass Spectrometry</a>
<ul>
<li class="chapter" data-level="10.1" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#commercial"><i class="fa fa-check"></i><b>10.1</b> Commercial</a>
<ul>
<li class="chapter" data-level="" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#raw-thermo"><i class="fa fa-check"></i>RAW (Thermo)</a></li>
<li class="chapter" data-level="" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#d-bruker-timstof-data"><i class="fa fa-check"></i>.d (Bruker) timsTOF data</a></li>
<li class="chapter" data-level="" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#wiff-sciex"><i class="fa fa-check"></i>WIFF (Sciex)</a></li>
<li class="chapter" data-level="" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#d-agilent"><i class="fa fa-check"></i>D (Agilent)</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#open-access-data"><i class="fa fa-check"></i><b>10.2</b> Open Access Data</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#proteowizard"><i class="fa fa-check"></i><b>10.2.1</b> ProteoWizard</a></li>
<li class="chapter" data-level="10.2.2" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#mzxml"><i class="fa fa-check"></i><b>10.2.2</b> mzXML</a></li>
<li class="chapter" data-level="10.2.3" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#mzml"><i class="fa fa-check"></i><b>10.2.3</b> mzML</a></li>
<li class="chapter" data-level="10.2.4" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#mzmlb"><i class="fa fa-check"></i><b>10.2.4</b> mzMLb</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#r-mass"><i class="fa fa-check"></i><b>10.3</b> R Mass</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#brain"><i class="fa fa-check"></i><b>10.3.1</b> BRAIN</a></li>
<li class="chapter" data-level="10.3.2" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#envipat"><i class="fa fa-check"></i><b>10.3.2</b> enviPat</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#r-ms-data"><i class="fa fa-check"></i><b>10.4</b> R MS Data</a>
<ul>
<li class="chapter" data-level="10.4.1" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#rformassspectrometry"><i class="fa fa-check"></i><b>10.4.1</b> RforMassSpectrometry</a></li>
<li class="chapter" data-level="10.4.2" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#msnbase"><i class="fa fa-check"></i><b>10.4.2</b> MSnbase</a></li>
<li class="chapter" data-level="10.4.3" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#mzr"><i class="fa fa-check"></i><b>10.4.3</b> mzR</a></li>
<li class="chapter" data-level="10.4.4" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#xcms"><i class="fa fa-check"></i><b>10.4.4</b> xcms</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#r-ms-analysis"><i class="fa fa-check"></i><b>10.5</b> R MS Analysis</a>
<ul>
<li class="chapter" data-level="10.5.1" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#msstats"><i class="fa fa-check"></i><b>10.5.1</b> MSstats</a></li>
<li class="chapter" data-level="10.5.2" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#tidyproteomics"><i class="fa fa-check"></i><b>10.5.2</b> Tidyproteomics</a></li>
<li class="chapter" data-level="10.5.3" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#assayr"><i class="fa fa-check"></i><b>10.5.3</b> AssayR</a></li>
<li class="chapter" data-level="10.5.4" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#deqms"><i class="fa fa-check"></i><b>10.5.4</b> DEqMS</a></li>
<li class="chapter" data-level="10.5.5" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#ms-empire"><i class="fa fa-check"></i><b>10.5.5</b> MS-EmpiRe</a></li>
<li class="chapter" data-level="10.5.6" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#msgrob2"><i class="fa fa-check"></i><b>10.5.6</b> msgrob2</a></li>
<li class="chapter" data-level="10.5.7" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#statspro"><i class="fa fa-check"></i><b>10.5.7</b> StatsPro</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="answers-to-exercises.html"><a href="answers-to-exercises.html"><i class="fa fa-check"></i>Answers to Exercises</a>
<ul>
<li class="chapter" data-level="" data-path="answers-to-exercises.html"><a href="answers-to-exercises.html#r-syntax-1"><i class="fa fa-check"></i><strong>4</strong> R Syntax</a></li>
<li class="chapter" data-level="" data-path="answers-to-exercises.html"><a href="answers-to-exercises.html#r-objects-1"><i class="fa fa-check"></i><strong>5</strong> R Objects</a></li>
<li class="chapter" data-level="" data-path="answers-to-exercises.html"><a href="answers-to-exercises.html#tidyverse-2"><i class="fa fa-check"></i><strong>6</strong> Tidyverse</a></li>
<li class="chapter" data-level="" data-path="answers-to-exercises.html"><a href="answers-to-exercises.html#data-wrangling-1"><i class="fa fa-check"></i><strong>7</strong> Data Wrangling</a></li>
<li class="chapter" data-level="" data-path="answers-to-exercises.html"><a href="answers-to-exercises.html#data-visualization-1"><i class="fa fa-check"></i><strong>8</strong> Data Visualization</a></li>
<li class="chapter" data-level="" data-path="answers-to-exercises.html"><a href="answers-to-exercises.html#more-5-r-objects"><i class="fa fa-check"></i>More <strong>5</strong> R Objects</a></li>
<li class="chapter" data-level="" data-path="answers-to-exercises.html"><a href="answers-to-exercises.html#more-6-tidyverse"><i class="fa fa-check"></i>More <strong>6</strong> Tidyverse</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/jeffsocal/asms_r_intro" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Learning R for Mass Spectrometry</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="mass-spectrometry" class="section level1 hasAnchor" number="10">
<h1><span class="header-section-number">Chapter 10</span> Mass Spectrometry<a href="mass-spectrometry.html#mass-spectrometry" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Mass spectrometry experiments produce a vast amount of data that comes in various formats. Mass spectrometry data is composed of raw data (m/z peak data) from different types of proprietary instrument vendor formats, open-access formats, and data processing and interpretation workflow formats such as XML, CSV, tab delimited, and Excel.</p>
<p>To handle such a diverse set of data, researchers can rely on multiple R packages that help access, manipulate, and process the data. These packages are designed to perform tasks such as filtering, normalization, and quality control, which are essential to produce reliable results.</p>
<p>Furthermore, the ability to analyze and interpret mass spectrometry data is essential for many fields, including proteomics, metabolomics, and lipidomics. Researchers use mass spectrometry data to identify and quantify proteins, metabolites, and lipids in biological samples.</p>
<p>The field of mass spectrometry generates a considerable amount of data in various formats that require processing and interpretation. With the help of a slew of R packages, researchers can efficiently manage and analyze this data to provide accurate and reliable results.</p>
<p><br />
</p>
<table>
<colgroup>
<col width="32%" />
<col width="67%" />
</colgroup>
<tbody>
<tr class="odd">
<td><img src="images/03.png" height="68" /></td>
<td><p><strong>At the end of this chapter you should be able to</strong></p>
<ul>
<li><p>Understand commercially proprietary raw data.</p></li>
<li><p>How to convert raw data to an open access format.</p></li>
<li><p>Be familiar with some R packages specific to mass spectrometry.</p></li>
</ul></td>
</tr>
</tbody>
</table>
<p><br />
</p>

<div id="commercial" class="section level2 hasAnchor" number="10.1">
<h2><span class="header-section-number">10.1</span> Commercial<a href="mass-spectrometry.html#commercial" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Accessing mass spectrometry data has some barriers due to vendors proprietary file formats. In recent years, vendors have been providing software development kits (SDKs) that allow programmers to provide direct access to the raw data. This is essential when developing fundamental workflows that require such data and also can enable analysis tools to bypass sometimes time consuming file format conversions to open formats. There are some means to access such raw data through R, allowing end-to-end pipelines to be developed directly from vendor data.</p>
<hr />
<div id="raw-thermo" class="section level3 unnumbered hasAnchor">
<h3>RAW (Thermo)<a href="mass-spectrometry.html#raw-thermo" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="rawrr" class="section level4 unnumbered hasAnchor">
<h4>rawrr<a href="mass-spectrometry.html#rawrr" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>This package provides access to proprietary Thermo Fisher Scientific Orbitrap instrument data as a stand-alone R package or serves as MsRawFileReaderBackend for the Bioconductor Spectra package. rawrr wraps the functionality of the RawFileReader .NET assembly. Test files are provided by the tartare ExperimentData package.<br />
</p>
<table>
<colgroup>
<col width="21%" />
<col width="78%" />
</colgroup>
<tbody>
<tr class="odd">
<td><img src="images/90_logo_rawrr.png" height="99" /></td>
<td><p><a href="https://bioconductor.org/packages/release/bioc/manuals/rawrr/man/rawrr.pdf">PDF manual</a></p>
<p><a href="https://bioconductor.org/packages/release/bioc/vignettes/rawrr/inst/doc/rawrr.html">WEB tutorial</a></p>
<p><a href="https://github.com/fgcz/rawrr">GitHub</a></p>
<p><a href="https://bioconductor.org/packages/release/bioc/html/rawrr.html">Bioconductor</a></p>
<p><em>Kockmann, Tobias, and Christian Panse. “The rawrr R package: direct access to Orbitrap data and beyond.” Journal of Proteome Research 20.4 (2021): 2028-2034.</em></p></td>
</tr>
</tbody>
</table>
<p><br />
</p>
</div>
<div id="rawreader" class="section level4 unnumbered hasAnchor">
<h4>rawReadeR<a href="mass-spectrometry.html#rawreader" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>This package is an R/C++ API to the MSFileReader.dll. rawReadeR allows for m/z profiles and scan header data to be extracted from .RAW files without the need for manual extraction using Xcalibur or conversion to a more universal format (.mzML, .mZXML, etc…)<br />
</p>
<table>
<colgroup>
<col width="40%" />
<col width="60%" />
</colgroup>
<tbody>
<tr class="odd">
<td><img src="images/05.png" height="68" /></td>
<td><a href="https://github.com/wilsontom/rawReadeR">GitHub</a></td>
</tr>
</tbody>
</table>
<p><br />
</p>
<hr />
</div>
</div>
<div id="d-bruker-timstof-data" class="section level3 unnumbered hasAnchor">
<h3>.d (Bruker) timsTOF data<a href="mass-spectrometry.html#d-bruker-timstof-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="timsr" class="section level4 unnumbered hasAnchor">
<h4>TimsR<a href="mass-spectrometry.html#timsr" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Bruker has released an SDK that has now been ported to python and R. It provides a simple way to get data out of results collected with your Bruker timsTOF Pro from R. This definitely ain’t no rocket science, but is pretty useful! The data is reported in stored in data.table objects, that are the only thing R has to actually work meaningfully with big data sets.</p>
<p><br />
</p>
<table>
<colgroup>
<col width="40%" />
<col width="59%" />
</colgroup>
<tbody>
<tr class="odd">
<td><img src="images/05.png" height="68" /></td>
<td><a href="https://github.com/MatteoLacki/timsr">GitHub</a></td>
</tr>
</tbody>
</table>
<p><br />
</p>
<hr />
</div>
</div>
<div id="wiff-sciex" class="section level3 unnumbered hasAnchor">
<h3>WIFF (Sciex)<a href="mass-spectrometry.html#wiff-sciex" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>None known at the moment.</p>
<hr />
</div>
<div id="d-agilent" class="section level3 unnumbered hasAnchor">
<h3>D (Agilent)<a href="mass-spectrometry.html#d-agilent" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Convert from raw using vendor software.</p>
<hr />

</div>
</div>
<div id="open-access-data" class="section level2 hasAnchor" number="10.2">
<h2><span class="header-section-number">10.2</span> Open Access Data<a href="mass-spectrometry.html#open-access-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Mass spectrometry (MS) is a powerful analytical method that can be used to determine the mass-to-charge ratio (m/z) of ions in a sample. Mass spectrometry data is generated from MS experiments and can be stored in various formats, including mzXML, mzML, and mzH5.</p>
<div id="proteowizard" class="section level3 hasAnchor" number="10.2.1">
<h3><span class="header-section-number">10.2.1</span> ProteoWizard<a href="mass-spectrometry.html#proteowizard" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>An open-source software suite that provides a collection of open-source, cross-platform software libraries and tools for extracting raw mass spectrometry data from from various instrument vendor formats and converting it to the formats listed below.</p>
<blockquote>
<p>_Kessner, D., Chambers, M., Burke, R., Agus, D., &amp; Mallick, P. (2008). ProteoWizard: open source software for rapid proteomics tools development. Bioinformatics, 24(21), 2534–2536.</p>
<p><a href="https://proteowizard.sourceforge.io/">proteowizard.sourceforge.io/</a></p>
</blockquote>
<hr />
</div>
<div id="mzxml" class="section level3 hasAnchor" number="10.2.2">
<h3><span class="header-section-number">10.2.2</span> mzXML<a href="mass-spectrometry.html#mzxml" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>mzXML is an open XML-based format for encoding MS data. It was developed by the Seattle Proteome Center and is widely used in the mass spectrometry community. mzXML files contain raw MS data, as well as metadata describing the instrument parameters used to acquire the data. mzXML files can be processed using a variety of software tools, such as the Trans-Proteomic Pipeline and ProteoWizard.</p>
<p>The mzXML format has been shown to be effective in handling data from a wide range of instruments. It has a simple structure that makes it easy to parse and process, making it an attractive choice for many researchers. The format is also relatively lightweight, which makes it easy to transfer and store large amounts of data.</p>
<blockquote>
<p><em>Pedrioli, P.G., Eng, J.K., Hubley, R., Vogelzang, M., Deutsch, E.W., Raught, B., Pratt, B., Nilsson, E., Angeletti, R.H., Apweiler, R. and Cheung, K., 2004. A common open representation of mass spectrometry data and its application to proteomics research. Nature biotechnology, 22(11), pp.1459-1466.</em></p>
</blockquote>
<hr />
</div>
<div id="mzml" class="section level3 hasAnchor" number="10.2.3">
<h3><span class="header-section-number">10.2.3</span> mzML<a href="mass-spectrometry.html#mzml" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The mzML format is another open XML-based format for MS data, developed by the Proteomics Standards Initiative. It is designed to be more flexible than mzXML and includes more detailed metadata. mzML files can be processed using software tools such as OpenMS and mzR.</p>
<p>The mzML format allows for more detailed and comprehensive data storage than mzXML. This is because mzML has a more complex structure, which enables the storage of a wider range of experimental metadata. The format is also more flexible, which means that it can be easily adapted to different types of experiments and instruments.</p>
<blockquote>
<p><em>Martens, L., Chambers, M., Sturm, M., Kessner, D., Levander, F., Shofstahl, J., Tang, W.H., Römpp, A., Neumann, S., Pizarro, A.D. and Montecchi-Palazzi, L., 2011. mzML—a community standard for mass spectrometry data. Molecular &amp; cellular proteomics, 10(1), p.R110. 000133.</em></p>
</blockquote>
<hr />
</div>
<div id="mzmlb" class="section level3 hasAnchor" number="10.2.4">
<h3><span class="header-section-number">10.2.4</span> mzMLb<a href="mass-spectrometry.html#mzmlb" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Recently proposed as a new file format based on HDF5 and NetCDF4 standards, mzMLb is faster and more flexible than existing approaches while preserving the XML encoding of metadata. Additionally, it is optimized for both read/write speed and storage efficiency. The format has a reference implementation provided within the ProteoWizard toolkit.</p>
<blockquote>
<p><em>Bhamber, Ranjeet S., et al. “mzMLb: A future-proof raw mass spectrometry data format based on standards-compliant mzML and optimized for speed and storage requirements.” Journal of proteome research 20.1 (2020): 172-183.</em></p>
</blockquote>
<hr />

</div>
</div>
<div id="r-mass" class="section level2 hasAnchor" number="10.3">
<h2><span class="header-section-number">10.3</span> R Mass<a href="mass-spectrometry.html#r-mass" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Molecular mass is the sum of the atomic masses of all the atoms in a molecule. It is an important parameter used in various fields of chemistry. The molecular mass of a molecule is usually expressed in atomic mass units (amu) or daltons (Da). Mass spectrometry based measurements require a charge and are expressed as mass-to-charge (m/z) or as Thompsons (Th).</p>
<p>Isotopic probabilities are also important in determining molecular mass. Isotopes are atoms of the same element that have different numbers of neutrons in their nuclei. Isotopes of an element have different atomic masses. The isotopic probability of an element is the probability that a given isotope of that element will occur in nature. For example, carbon (C) has two stable isotopes, carbon-12 (12C) and carbon-13 (13C), with atomic masses of 12.000 amu and 13.003 amu, respectively. The isotopic probability of carbon-12 is 98.9%, while that of carbon-13 is 1.1%. Therefore, the average atomic mass of carbon in nature is 12.011 amu.</p>
<p>There are several measures of a molecular mass that can be reported. The <em>average mass</em> is that of the weighted average of all isotopes and likely to be reported for small molecules. While the <em>nominal mass</em> of a molecule is defined as the sum of the integer masses of the most abundant isotopes in a molecule. The <em>monoisotopic mass</em> is commonly considered the as the sum of the exact masses of the lightest isotopes, and this value is considered in all peptide-based proteomics applications.</p>
<hr />

<div id="brain" class="section level3 hasAnchor" number="10.3.1">
<h3><span class="header-section-number">10.3.1</span> BRAIN<a href="mass-spectrometry.html#brain" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Baffling Recursive Algorithm for Isotope distributioN calculations</strong></p>
<p>R BRAIN is an isotopic abundance calculator implemented in R programming language and is especially useful for chemists and researchers who deal with complex molecules and need to calculate their isotopic composition accurately. In addition, it has a handy function for calculating the mass directly from an amino acid sequence.</p>
<p><br />
</p>
<table>
<colgroup>
<col width="15%" />
<col width="84%" />
</colgroup>
<tbody>
<tr class="odd">
<td><img src="images/05.png" height="68" /></td>
<td><p><a href="https://bioconductor.org/packages/release/bioc/manuals/BRAIN/man/BRAIN.pdf">Bioconductor</a></p>
<p><em>Dittwald, Piotr, et al. “BRAIN: a universal tool for high-throughput calculations of the isotopic distribution for mass spectrometry.” Analytical chemistry 85.4 (2013): 1991-1994.</em></p></td>
</tr>
</tbody>
</table>
<p><br />
</p>
<div id="installation-1" class="section level4 unnumbered hasAnchor">
<h4>Installation<a href="mass-spectrometry.html#installation-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb278"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb278-1"><a href="mass-spectrometry.html#cb278-1" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;BRAIN&quot;</span>)</span></code></pre></div>
</div>
<div id="useage" class="section level4 unnumbered hasAnchor">
<h4>Useage<a href="mass-spectrometry.html#useage" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb279"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb279-1"><a href="mass-spectrometry.html#cb279-1" tabindex="-1"></a><span class="fu">library</span>(BRAIN)</span>
<span id="cb279-2"><a href="mass-spectrometry.html#cb279-2" tabindex="-1"></a></span>
<span id="cb279-3"><a href="mass-spectrometry.html#cb279-3" tabindex="-1"></a><span class="co"># Human insulin amino acid sequence</span></span>
<span id="cb279-4"><a href="mass-spectrometry.html#cb279-4" tabindex="-1"></a>str_seq <span class="ot">&lt;-</span> <span class="st">&quot;MALWMRLLPLLALLALWGPDPAAAFVNQHLCGSHLVEALYLVCGERGFFYTPKTRREAEDLQVGQVELGGGPGAGSLQPLALEGSLQKRGIVEQCCTSICSLYQLENYCN&quot;</span></span>
<span id="cb279-5"><a href="mass-spectrometry.html#cb279-5" tabindex="-1"></a></span>
<span id="cb279-6"><a href="mass-spectrometry.html#cb279-6" tabindex="-1"></a><span class="co"># get a list-object of atoms</span></span>
<span id="cb279-7"><a href="mass-spectrometry.html#cb279-7" tabindex="-1"></a>lst_atm <span class="ot">&lt;-</span> <span class="fu">getAtomsFromSeq</span>(str_seq) </span></code></pre></div>
<p>Calculate the average mass.</p>
<div class="sourceCode" id="cb280"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb280-1"><a href="mass-spectrometry.html#cb280-1" tabindex="-1"></a><span class="fu">calculateAverageMass</span>(lst_atm)</span></code></pre></div>
<pre><code>## [1] 11980.82</code></pre>
<p>Calculate the monoisotopic mass.</p>
<div class="sourceCode" id="cb282"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb282-1"><a href="mass-spectrometry.html#cb282-1" tabindex="-1"></a><span class="fu">calculateMonoisotopicMass</span>(lst_atm)</span></code></pre></div>
<pre><code>## [1] 11973.02</code></pre>
<p>Calculate the isotopic abundances (probabilities) mass.</p>
<div class="sourceCode" id="cb284"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb284-1"><a href="mass-spectrometry.html#cb284-1" tabindex="-1"></a>lst_isotopes <span class="ot">&lt;-</span> <span class="fu">useBRAIN</span>(lst_atm, <span class="at">nrPeaks =</span> <span class="dv">20</span>)</span>
<span id="cb284-2"><a href="mass-spectrometry.html#cb284-2" tabindex="-1"></a></span>
<span id="cb284-3"><a href="mass-spectrometry.html#cb284-3" tabindex="-1"></a><span class="fu">plot</span>(lst_isotopes<span class="sc">$</span>masses, lst_isotopes<span class="sc">$</span>isoDistr, <span class="at">xlab=</span><span class="st">&#39;Mass&#39;</span>, <span class="at">ylab=</span><span class="st">&#39;Probability&#39;</span>, <span class="at">type =</span> <span class="st">&#39;h&#39;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-229-1.png" width="576" style="display: block; margin: auto;" />
Calculate the isotopic abundances (probabilities) mass for a metabolite [C100H200S2Cl5].</p>
<div class="sourceCode" id="cb285"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb285-1"><a href="mass-spectrometry.html#cb285-1" tabindex="-1"></a>lst_atm <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">C=</span><span class="dv">100</span>, <span class="at">H=</span><span class="dv">200</span>, <span class="at">S=</span><span class="dv">2</span>, <span class="at">Cl=</span><span class="dv">5</span>)</span>
<span id="cb285-2"><a href="mass-spectrometry.html#cb285-2" tabindex="-1"></a></span>
<span id="cb285-3"><a href="mass-spectrometry.html#cb285-3" tabindex="-1"></a>lst_isotopes <span class="ot">&lt;-</span> <span class="fu">useBRAIN</span>(lst_atm, <span class="at">nrPeaks =</span> <span class="dv">13</span>)</span>
<span id="cb285-4"><a href="mass-spectrometry.html#cb285-4" tabindex="-1"></a></span>
<span id="cb285-5"><a href="mass-spectrometry.html#cb285-5" tabindex="-1"></a><span class="fu">plot</span>(<span class="at">x =</span> lst_isotopes<span class="sc">$</span>masses, </span>
<span id="cb285-6"><a href="mass-spectrometry.html#cb285-6" tabindex="-1"></a>     <span class="at">y =</span> lst_isotopes<span class="sc">$</span>isoDistr <span class="sc">/</span> <span class="fu">max</span>(lst_isotopes<span class="sc">$</span>isoDistr) <span class="sc">*</span> <span class="dv">100</span>, </span>
<span id="cb285-7"><a href="mass-spectrometry.html#cb285-7" tabindex="-1"></a>     <span class="at">xlab=</span><span class="st">&#39;Mass&#39;</span>, <span class="at">ylab=</span><span class="st">&#39;Relative abundance&#39;</span>, <span class="at">type =</span> <span class="st">&#39;h&#39;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-230-1.png" width="576" style="display: block; margin: auto;" /></p>
<hr />

</div>
</div>
<div id="envipat" class="section level3 hasAnchor" number="10.3.2">
<h3><span class="header-section-number">10.3.2</span> enviPat<a href="mass-spectrometry.html#envipat" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Isotope Pattern, Profile and Centroid Calculation for Mass Spectrometry</strong></p>
<p>A method for calculating theoretical isotope patterns in mass spectrometry. This method uses a treelike structure to derive sets of subisotopologues for each element in a molecule. By doing so, it allows for early pruning of low-probability isotopologues and the detection of the most probable isotopologue. The method was validated in a large-scale benchmark simulation.</p>
<p><br />
</p>
<table>
<colgroup>
<col width="18%" />
<col width="81%" />
</colgroup>
<tbody>
<tr class="odd">
<td><img src="images/05.png" height="68" /></td>
<td><p><a href="https://cran.r-project.org/web/packages/enviPat/enviPat.pdf">CRAN</a></p>
<p><em>Loos, Martin, et al. “Accelerated isotope fine structure calculation using pruned transition trees.” Analytical chemistry 87.11 (2015): 5738-5744.</em></p></td>
</tr>
</tbody>
</table>
<p><br />
</p>
<div id="installation-2" class="section level4 unnumbered hasAnchor">
<h4>Installation<a href="mass-spectrometry.html#installation-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb286"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb286-1"><a href="mass-spectrometry.html#cb286-1" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;enviPat&quot;</span>)</span></code></pre></div>
</div>
<div id="use" class="section level4 unnumbered hasAnchor">
<h4>Use<a href="mass-spectrometry.html#use" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb287"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb287-1"><a href="mass-spectrometry.html#cb287-1" tabindex="-1"></a><span class="fu">library</span>(enviPat)</span>
<span id="cb287-2"><a href="mass-spectrometry.html#cb287-2" tabindex="-1"></a></span>
<span id="cb287-3"><a href="mass-spectrometry.html#cb287-3" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;isotopes&quot;</span>)</span>
<span id="cb287-4"><a href="mass-spectrometry.html#cb287-4" tabindex="-1"></a></span>
<span id="cb287-5"><a href="mass-spectrometry.html#cb287-5" tabindex="-1"></a>pattern <span class="ot">&lt;-</span> <span class="fu">isopattern</span>(</span>
<span id="cb287-6"><a href="mass-spectrometry.html#cb287-6" tabindex="-1"></a>  isotopes,</span>
<span id="cb287-7"><a href="mass-spectrometry.html#cb287-7" tabindex="-1"></a>  <span class="st">&quot;C100H200S2Cl5&quot;</span>,</span>
<span id="cb287-8"><a href="mass-spectrometry.html#cb287-8" tabindex="-1"></a>  <span class="at">threshold=</span><span class="fl">0.1</span>,</span>
<span id="cb287-9"><a href="mass-spectrometry.html#cb287-9" tabindex="-1"></a>  <span class="at">plotit=</span><span class="cn">TRUE</span>,</span>
<span id="cb287-10"><a href="mass-spectrometry.html#cb287-10" tabindex="-1"></a>  <span class="at">charge=</span><span class="cn">FALSE</span>,</span>
<span id="cb287-11"><a href="mass-spectrometry.html#cb287-11" tabindex="-1"></a>  <span class="at">emass=</span><span class="fl">0.00054858</span>,</span>
<span id="cb287-12"><a href="mass-spectrometry.html#cb287-12" tabindex="-1"></a>  <span class="at">algo=</span><span class="dv">1</span></span>
<span id="cb287-13"><a href="mass-spectrometry.html#cb287-13" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-232-1.png" width="576" style="display: block; margin: auto;" /></p>

</div>
</div>
</div>
<div id="r-ms-data" class="section level2 hasAnchor" number="10.4">
<h2><span class="header-section-number">10.4</span> R MS Data<a href="mass-spectrometry.html#r-ms-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="rformassspectrometry" class="section level3 hasAnchor" number="10.4.1">
<h3><span class="header-section-number">10.4.1</span> RforMassSpectrometry<a href="mass-spectrometry.html#rformassspectrometry" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The RforMassSpectrometry initiative is a collaborative project aimed at developing efficient, documented, and flexible R software for analyzing high throughput mass spectrometry assays. The project formalizes the long-time collaborative development efforts of its core members under the RforMassSpectrometry organization to facilitate dissemination and accessibility of their work.</p>
<p><br />
</p>
<table>
<colgroup>
<col width="35%" />
<col width="64%" />
</colgroup>
<tbody>
<tr class="odd">
<td><img src="images/90_logo_rformassspec.png" height="99" /></td>
<td><p><a href="https://rformassspectrometry.github.io/docs/">WEB tutorial</a></p>
<p><a href="https://www.rformassspectrometry.org/pkgs/">Packages</a></p></td>
</tr>
</tbody>
</table>
<p><br />
</p>
<p>The suite of packages developed by the initiative is available on GitHub under the RforMassSpectrometry organization. This platform serves as the main development and collaboration area of the project. Once the packages have reached a satisfactory level of maturity and stability, they are submitted to the Bioconductor project to integrate with existing and broader infrastructure for high throughput biology data.</p>
<p>The RforMassSpectrometry initiative focuses on co-developing and maintaining interoperable software to tackle a wide range of needs in computational mass spectrometry using the R language and environment. The goal is not to address all possible needs, but rather to provide a set of software that work together efficiently, rather than maximizing the number of packages.</p>
<div id="packages" class="section level4 unnumbered hasAnchor">
<h4>Packages<a href="mass-spectrometry.html#packages" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div id="msexperiment" class="section level5 unnumbered hasAnchor">
<h5>MsExperiment<a href="mass-spectrometry.html#msexperiment" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>The MsExperiment package provides the infrastructure to store and manage all aspects related to a complete proteomics or metabolomics mass spectrometry experiment. It relies on the other RforMassSpectrometry core packages for the data crunching.</p>
</div>
<div id="spectra" class="section level5 unnumbered hasAnchor">
<h5>Spectra<a href="mass-spectrometry.html#spectra" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>The Spectra package provides base classes and processing methods for raw mass spectrometry data. It is designed with efficiency, both in terms of memory footprint and processing time in mind, and can manage data in different types of formats.</p>
</div>
<div id="qfeatures" class="section level5 unnumbered hasAnchor">
<h5>QFeatures<a href="mass-spectrometry.html#qfeatures" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>The QFeatures package offers the infrastructure to manage and process quantitative features for high-throughput mass spectrometry assays, including proteomics and metabolomics experiments.</p>
</div>
<div id="psmatch" class="section level5 unnumbered hasAnchor">
<h5>PSMatch<a href="mass-spectrometry.html#psmatch" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>The PSMatch package allows to read, process and analyse peptide-spectrum matches.</p>
</div>
<div id="chromatograms" class="section level5 unnumbered hasAnchor">
<h5>Chromatograms<a href="mass-spectrometry.html#chromatograms" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>The Chromatograms package provides base classes and processing methods for chromatographic data. It is designed with efficiency, both in terms of memory footprint and processing time in mind, and can manage data in different types of formats.</p>
</div>
<div id="mscoreutils" class="section level5 unnumbered hasAnchor">
<h5>MsCoreUtils<a href="mass-spectrometry.html#mscoreutils" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>The MsCoreUtils package defines low-level functions for mass spectrometry data processing and is independent of any high-level data structures.</p>
</div>
<div id="metabocoreutils" class="section level5 unnumbered hasAnchor">
<h5>MetaboCoreUtils<a href="mass-spectrometry.html#metabocoreutils" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>The MetaboCoreUtils package defines low-level functions for common operations in metabolomics and is independent of any high-level data structures.</p>
<hr />

</div>
</div>
</div>
<div id="msnbase" class="section level3 hasAnchor" number="10.4.2">
<h3><span class="header-section-number">10.4.2</span> MSnbase<a href="mass-spectrometry.html#msnbase" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Often, the full access to raw data provides a complete sense of control. However, there are times when attempting to achieve mundane tasks can be extremely tedious and repetitive. This is where the concept of abstraction comes in. It is unnecessary to be aware of every detail that is exposed by mzR to work with and manipulate raw data. This is where <code>MSnbase</code> and the <code>MSnExp</code> data structure come in. These tools provide a smoother and more efficient approach to handle and annotate raw data.</p>
<p>While MSnbase only provides rudimentary access to the data, there are many other algorithms that add additional functionality such as xcms, which provides powerful visualization tools for mass spectrometry data.</p>
<p><br />
</p>
<table>
<colgroup>
<col width="14%" />
<col width="85%" />
</colgroup>
<tbody>
<tr class="odd">
<td><img src="images/05.png" height="68" /></td>
<td><p><a href="https://bioconductor.org/packages/release/bioc/vignettes/MSnbase/inst/doc/v05-MSnbase-development.html">WEB tutorial</a></p>
<p><a href="https://bioconductor.org/packages/release/bioc/manuals/MSnbase/man/MSnbase.pdf">PDF manual</a></p>
<p><a href="https://bioconductor.org/packages/release/bioc/html/MSnbase.html">Bioconductor</a></p>
<p><em>Gatto, Laurent, and Kathryn S. Lilley. “MSnbase-an R/Bioconductor package for isobaric tagged mass spectrometry data visualization, processing and quantitation.” Bioinformatics 28.2 (2012): 288-289.</em></p></td>
</tr>
</tbody>
</table>
<p><br />
</p>
<p><strong>Note</strong> dependencies for MSnbase are not available for Mac M1 cpus.</p>
<hr />

</div>
<div id="mzr" class="section level3 hasAnchor" number="10.4.3">
<h3><span class="header-section-number">10.4.3</span> mzR<a href="mass-spectrometry.html#mzr" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The <code>mzR</code> package is a powerful tool for analyzing large raw data files in proteomics research. It offers a direct interface to the <a href="http://proteowizard.sourceforge.net/">proteowizard</a> code base, which is a widely-used software framework for mass spectrometry data analysis. By leveraging a substantial proportion of <em>pwiz</em>’s C/C++ code, <code>mzR</code> is able to provide fast and efficient parsing of these complex data files.</p>
<p><br />
</p>
<table>
<colgroup>
<col width="19%" />
<col width="80%" />
</colgroup>
<tbody>
<tr class="odd">
<td><img src="images/05.png" height="68" /></td>
<td><p><a href="https://lgatto.github.io/2020-02-17-RProt-Berlin/raw-ms-data-mzr-and-msnbase.html">WEB tutorial</a></p>
<p><a href="https://bioconductor.org/packages/release/bioc/manuals/mzR/man/mzR.pdf">PDF manual</a></p>
<p><a href="https://bioconductor.org/packages/release/bioc/html/mzR.html">Bioconductor</a></p>
<p><em>Chambers, Matthew C., et al. “A cross-platform toolkit for mass spectrometry and proteomics.” Nature biotechnology 30.10 (2012): 918-920.</em></p></td>
</tr>
</tbody>
</table>
<p><br />
</p>
<p>Read in an <a href="mass-spectrometry.html#mzml">10.2.3</a> mzML converted LCMS data file and check its contents.</p>
<div class="sourceCode" id="cb288"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb288-1"><a href="mass-spectrometry.html#cb288-1" tabindex="-1"></a><span class="fu">library</span>(mzR)</span>
<span id="cb288-2"><a href="mass-spectrometry.html#cb288-2" tabindex="-1"></a></span>
<span id="cb288-3"><a href="mass-spectrometry.html#cb288-3" tabindex="-1"></a>ms_dat <span class="ot">&lt;-</span> <span class="fu">openMSfile</span>(<span class="st">&quot;~/Local/QC_20230214_QE_Hela200ng_Aur2hr.mzML&quot;</span>)</span>
<span id="cb288-4"><a href="mass-spectrometry.html#cb288-4" tabindex="-1"></a></span>
<span id="cb288-5"><a href="mass-spectrometry.html#cb288-5" tabindex="-1"></a>ms_dat</span></code></pre></div>
<pre><code>## Mass Spectrometry file handle.
## Filename:  QC_20230214_QE_Hela200ng_Aur2hr.mzML 
## Number of scans:  12853</code></pre>
<p>Extract out a specific spectrum.</p>
<div class="sourceCode" id="cb290"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb290-1"><a href="mass-spectrometry.html#cb290-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb290-2"><a href="mass-spectrometry.html#cb290-2" tabindex="-1"></a></span>
<span id="cb290-3"><a href="mass-spectrometry.html#cb290-3" tabindex="-1"></a>tbl_spec <span class="ot">&lt;-</span> <span class="fu">peaks</span>(ms_dat, <span class="dv">1234</span>) <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>()</span>
<span id="cb290-4"><a href="mass-spectrometry.html#cb290-4" tabindex="-1"></a></span>
<span id="cb290-5"><a href="mass-spectrometry.html#cb290-5" tabindex="-1"></a>tbl_spec</span></code></pre></div>
<pre><code>## # A tibble: 732 × 2
##       mz intensity
##    &lt;dbl&gt;     &lt;dbl&gt;
##  1  375.    33308.
##  2  375.   915292.
##  3  376.    23732.
##  4  376.   159740.
##  5  376.    25760.
##  6  376.   184179.
##  7  377.    30452.
##  8  377.    69894.
##  9  377.    15017.
## 10  377.   822320.
## # … with 722 more rows</code></pre>
<div class="sourceCode" id="cb292"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb292-1"><a href="mass-spectrometry.html#cb292-1" tabindex="-1"></a>tbl_spec <span class="sc">%&gt;%</span> </span>
<span id="cb292-2"><a href="mass-spectrometry.html#cb292-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(mz, intensity)) <span class="sc">+</span> </span>
<span id="cb292-3"><a href="mass-spectrometry.html#cb292-3" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">xend =</span> mz, <span class="at">yend =</span> <span class="dv">0</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-236-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>In addition to its parsing capabilities, <code>mzR</code> includes a range of useful functions for working with mass spectrometry data. This package supports a wide variety of file formats, and provides functions for reading, writing, and manipulating data in these formats. Furthermore, <code>mzR</code> includes features for working with tandem mass spectrometry (MS/MS) data, including functions for spectral processing and peak picking.</p>
<hr />

</div>
<div id="xcms" class="section level3 hasAnchor" number="10.4.4">
<h3><span class="header-section-number">10.4.4</span> xcms<a href="mass-spectrometry.html#xcms" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>XCMS is an R package that provides a collection of functions for processing and analyzing mass spectrometry (MS) data. This package is designed to extract meaningful information from large data sets generated by MS-based experiments, such as metabolomics and proteomics studies. The web tutorials offer an excellent starting point.</p>
<p><strong>Note</strong> dependencies for xcms are not available for Mac M1 cpus.</p>
<p><br />
</p>
<table>
<colgroup>
<col width="14%" />
<col width="85%" />
</colgroup>
<tbody>
<tr class="odd">
<td><img src="images/05.png" height="68" /></td>
<td><p><a href="https://bioconductor.org/packages/release/bioc/vignettes/xcms/inst/doc/xcms-lcms-ms.html">WEB tutorial</a></p>
<p><a href="https://bioconductor.org/packages/3.12/bioc/manuals/xcms/man/xcms.pdf">PDF manual</a></p>
<p><a href="https://bioconductor.org/packages/3.12/bioc/html/xcms.html">Bioconductor</a></p>
<p><em>Smith, Colin A., et al. “XCMS: processing mass spectrometry data for metabolite profiling using nonlinear peak alignment, matching, and identification.” Analytical chemistry 78.3 (2006): 779-787.</em></p>
<p><em>Benton, H. Paul, et al. “XCMS2: processing tandem mass spectrometry data for metabolite identification and structural characterization.” Analytical chemistry 80.16 (2008): 6382-6389.</em></p></td>
</tr>
</tbody>
</table>
<p><br />
</p>
<p>XCMS offers a wide range of functionalities that are necessary for preprocessing and analyzing raw MS data. Some of the key features of XCMS include:</p>
<ul>
<li>Data preprocessing: The package offers a variety of preprocessing functions, such as peak detection, retention time correction, and noise filtering.</li>
<li>Feature detection: XCMS provides functions for detecting and aligning features across multiple samples, which is essential for comparative analysis.</li>
<li>Statistical analysis: The package also includes functions for statistical analysis, such as principal component analysis (PCA), hierarchical clustering, and differential expression analysis.</li>
<li>Visualization: XCMS provides several visualization tools for exploring and interpreting MS data, including heatmaps, scatter plots, and volcano plots.</li>
</ul>
<p>Read in an <a href="mass-spectrometry.html#mzml">10.2.3</a> mzML converted LCMS data file and check its contents.</p>
<div class="sourceCode" id="cb293"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb293-1"><a href="mass-spectrometry.html#cb293-1" tabindex="-1"></a><span class="fu">library</span>(xcms)</span>
<span id="cb293-2"><a href="mass-spectrometry.html#cb293-2" tabindex="-1"></a></span>
<span id="cb293-3"><a href="mass-spectrometry.html#cb293-3" tabindex="-1"></a>ms_dat <span class="ot">&lt;-</span> <span class="fu">readMSData</span>(<span class="st">&quot;~/Local/gitdata/QC_20230214_QE_Hela200ng_Aur2hr.mzML&quot;</span>)</span></code></pre></div>
<p>Use a peak-detection algorithm to find chromatographic peaks for each precursor</p>
<div class="sourceCode" id="cb294"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb294-1"><a href="mass-spectrometry.html#cb294-1" tabindex="-1"></a>cwp <span class="ot">&lt;-</span> <span class="fu">CentWaveParam</span>(<span class="at">snthresh =</span> <span class="dv">5</span>, <span class="at">noise =</span> <span class="dv">100</span>, <span class="at">ppm =</span> <span class="dv">10</span>, <span class="at">peakwidth =</span> <span class="fu">c</span>(<span class="dv">3</span>, <span class="dv">30</span>))</span>
<span id="cb294-2"><a href="mass-spectrometry.html#cb294-2" tabindex="-1"></a></span>
<span id="cb294-3"><a href="mass-spectrometry.html#cb294-3" tabindex="-1"></a>ms_dat <span class="ot">&lt;-</span> <span class="fu">findChromPeaks</span>(ms_dat, <span class="at">param =</span> cwp)</span></code></pre></div>
<p>One of the main advantages of XCMS is its flexibility and scalability. This package is highly customizable, allowing users to tailor the analysis pipeline to their specific needs. In addition, XCMS is designed to handle large data sets, making it suitable for high-throughput experiments.</p>
<p>Currently, spectra data representation, handling and processing is being re-implemented as part of the <a href="mass-spectrometry.html#rformassspectrometry">10.4.1</a> RforMassSpectrometry initiative aiming at increasing the performance of methods and simplifying their use.</p>
<hr />

</div>
</div>
<div id="r-ms-analysis" class="section level2 hasAnchor" number="10.5">
<h2><span class="header-section-number">10.5</span> R MS Analysis<a href="mass-spectrometry.html#r-ms-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Finally, there are numerous R packages that implement methods, common and advanced, in statistical post analysis of quantitative data. Explored here are only a few of the available packages.</p>

<div id="msstats" class="section level3 hasAnchor" number="10.5.1">
<h3><span class="header-section-number">10.5.1</span> MSstats<a href="mass-spectrometry.html#msstats" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The MSstats package is an R package designed for the analysis of label-free mass spectrometry data. It provides a wide range of statistical tools for the analysis of protein abundance data, including normalization, missing value imputation, quality control, and differential expression analysis. MSstats provides a powerful and flexible way to analyze mass spectrometry data, making it an essential tool for researchers in the field.</p>
<p><br />
</p>
<table>
<colgroup>
<col width="16%" />
<col width="83%" />
</colgroup>
<tbody>
<tr class="odd">
<td><img src="images/05.png" height="68" /></td>
<td><p><a href="https://msstats.org">WEB tutorial</a></p>
<p><a href="https://bioconductor.org/packages/release/bioc/manuals/MSstats/man/MSstats.pdf">PDF manual</a></p>
<p><a href="https://www.bioconductor.org/packages/release/bioc/html/MSstats.html">Bioconductor</a></p>
<p><em>Choi, Meena, et al. “MSstats: an R package for statistical analysis of quantitative mass spectrometry-based proteomic experiments.” Bioinformatics 30.17 (2014): 2524-2526.</em></p></td>
</tr>
</tbody>
</table>
<p><br />
</p>
<hr />

</div>
<div id="tidyproteomics" class="section level3 hasAnchor" number="10.5.2">
<h3><span class="header-section-number">10.5.2</span> Tidyproteomics<a href="mass-spectrometry.html#tidyproteomics" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The tidyproteomics R package is a tool that provides a set of functions to preprocess and analyze proteomics data using the tidy data framework. This package is built on top of the tidyverse and Bioconductor packages, which are widely used in the R community for data manipulation and analysis.</p>
<p><br />
</p>
<table>
<colgroup>
<col width="44%" />
<col width="55%" />
</colgroup>
<tbody>
<tr class="odd">
<td><img src="images/90_logo_tidyproteomics.png" height="99" /></td>
<td><p><a href="https://jeffsocal.github.io/tidyproteomics/">WEB manual</a></p>
<p><a href="https://github.com/jeffsocal/tidyproteomics">GitHub</a></p>
<p><a href="https://bioinformatics.pel.caltech.edu/tidyproteomics/">WEB App</a></p></td>
</tr>
</tbody>
</table>
<p><br />
</p>
<p>Some of the main features of the tidyproteomics package include:</p>
<ul>
<li>Data preprocessing functions for common tasks such as filtering, normalization, and imputation.</li>
<li>Functions for quality assessment and visualization of proteomics data.</li>
<li>Integration with other Bioconductor packages for downstream analysis such as differential expression analysis and pathway analysis.</li>
</ul>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:10001"></span>
<img src="images/10_001_tidyproteomics.png" alt="\label{fig:10001}tidyproteomics workflow" width="440" />
<p class="caption">
Figure 10.1: tidyproteomics workflow
</p>
</div>
<div id="installation-3" class="section level4 unnumbered hasAnchor">
<h4>Installation<a href="mass-spectrometry.html#installation-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>To install the <code>tidyproteomics</code> package, you will need to install GitHub and Bioconductor repositories:</p>
<div class="sourceCode" id="cb295"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb295-1"><a href="mass-spectrometry.html#cb295-1" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;devtools&quot;</span>)</span>
<span id="cb295-2"><a href="mass-spectrometry.html#cb295-2" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;jeffsocal/tidyproteomics&quot;</span>)</span>
<span id="cb295-3"><a href="mass-spectrometry.html#cb295-3" tabindex="-1"></a></span>
<span id="cb295-4"><a href="mass-spectrometry.html#cb295-4" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb295-5"><a href="mass-spectrometry.html#cb295-5" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="fu">c</span>(<span class="st">&quot;limma&quot;</span>,<span class="st">&quot;qvalue&quot;</span>,<span class="st">&quot;fgsea&quot;</span>,<span class="st">&quot;Biostrings&quot;</span>))</span></code></pre></div>
</div>
<div id="loading-data" class="section level4 unnumbered hasAnchor">
<h4>Loading Data<a href="mass-spectrometry.html#loading-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>To load your data into tidyproteomics, you can use the following code:</p>
<div class="sourceCode" id="cb296"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb296-1"><a href="mass-spectrometry.html#cb296-1" tabindex="-1"></a><span class="co"># Load the tidyproteomics package</span></span>
<span id="cb296-2"><a href="mass-spectrometry.html#cb296-2" tabindex="-1"></a><span class="fu">library</span>(tidyproteomics)</span>
<span id="cb296-3"><a href="mass-spectrometry.html#cb296-3" tabindex="-1"></a></span>
<span id="cb296-4"><a href="mass-spectrometry.html#cb296-4" tabindex="-1"></a><span class="co"># Import data</span></span>
<span id="cb296-5"><a href="mass-spectrometry.html#cb296-5" tabindex="-1"></a>data_proteins <span class="ot">&lt;-</span> <span class="st">&quot;path_to_data.xlsx&quot;</span> <span class="sc">%&gt;%</span></span>
<span id="cb296-6"><a href="mass-spectrometry.html#cb296-6" tabindex="-1"></a>  <span class="fu">import</span>(<span class="st">&quot;ProteomeDiscoverer&quot;</span>, <span class="st">&quot;proteins&quot;</span>) </span></code></pre></div>
</div>
<div id="data-summaries" class="section level4 unnumbered hasAnchor">
<h4>Data Summaries<a href="mass-spectrometry.html#data-summaries" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Currently, tidyproteomics implements two summary quantitative visualizations. The first is a simple grouped bar chart that displays individual and grouped proteins, as well as all and unique peptides. The match-between-runs is shown as a margin above the MS2 evidenced identifications. In recent literature, a summary of protein quantitation has been visualized as a rank-based dot plot. This plot can be extended to highlight statistical differences via an unbiased all-pair-wise comparison, which gives an anticipated view on how to guide downstream analyses.</p>
<div class="sourceCode" id="cb297"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb297-1"><a href="mass-spectrometry.html#cb297-1" tabindex="-1"></a>p01 <span class="ot">&lt;-</span> data_proteins <span class="sc">%&gt;%</span> <span class="fu">plot_counts</span>()</span>
<span id="cb297-2"><a href="mass-spectrometry.html#cb297-2" tabindex="-1"></a>p02 <span class="ot">&lt;-</span> data_proteins <span class="sc">%&gt;%</span> <span class="fu">plot_quantrank</span>()</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-243-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>Summary Stats {-}</p>
<p>Summarizing proteomics data is vital to understanding the bigger picture and conveying summary stats that set the tone for the larger analysis. The results of each summary can be directed to via the destination option to “print” on screen, “save” to a file or “return” as a tibble.</p>
<div class="sourceCode" id="cb298"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb298-1"><a href="mass-spectrometry.html#cb298-1" tabindex="-1"></a>data_proteins <span class="ot">&lt;-</span> data_proteins <span class="sc">%&gt;%</span></span>
<span id="cb298-2"><a href="mass-spectrometry.html#cb298-2" tabindex="-1"></a>  <span class="co"># save a table of simple summary stats</span></span>
<span id="cb298-3"><a href="mass-spectrometry.html#cb298-3" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="st">&quot;sample&quot;</span>, <span class="at">destination =</span> <span class="st">&quot;save&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb298-4"><a href="mass-spectrometry.html#cb298-4" tabindex="-1"></a>  <span class="co"># save a report on contamination</span></span>
<span id="cb298-5"><a href="mass-spectrometry.html#cb298-5" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">contamination =</span> <span class="st">&quot;CRAP&quot;</span>, <span class="at">destination =</span> <span class="st">&quot;save&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb298-6"><a href="mass-spectrometry.html#cb298-6" tabindex="-1"></a>  <span class="co"># remove contamination</span></span>
<span id="cb298-7"><a href="mass-spectrometry.html#cb298-7" tabindex="-1"></a>  <span class="fu">subset</span>(<span class="sc">!</span>description <span class="sc">%like%</span> <span class="st">&quot;^CRAP&quot;</span>)</span></code></pre></div>
</div>
<div id="normalization-and-imputation" class="section level4 unnumbered hasAnchor">
<h4>Normalization and Imputation<a href="mass-spectrometry.html#normalization-and-imputation" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Quantitative proteomics requires accurate normalization, which can be difficult to implement. The <code>normalize()</code> function in the tidyproteomics package is a wrapper for various normalization methods, while <code>select_normalization()</code> automatically selects the best method based on a weighted score. Both functions allow for downstream analyses such as <code>expression()</code> and <code>enrichment()</code>. The package attempts to apply each function universally to peptide and protein values using the <code>identifier</code> variable to identify the thing being measured.</p>
<div class="sourceCode" id="cb299"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb299-1"><a href="mass-spectrometry.html#cb299-1" tabindex="-1"></a>data_proteins <span class="ot">&lt;-</span> data_proteins <span class="sc">%&gt;%</span></span>
<span id="cb299-2"><a href="mass-spectrometry.html#cb299-2" tabindex="-1"></a>  <span class="co"># normalize via several methods, best method will be automatically selected</span></span>
<span id="cb299-3"><a href="mass-spectrometry.html#cb299-3" tabindex="-1"></a>  <span class="fu">normalize</span>(<span class="at">.method =</span> <span class="fu">c</span>(<span class="st">&quot;median&quot;</span>,<span class="st">&quot;linear&quot;</span>,<span class="st">&quot;limma&quot;</span>,<span class="st">&quot;randomforest&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb299-4"><a href="mass-spectrometry.html#cb299-4" tabindex="-1"></a>  <span class="co"># impute with a minimum value (this is a knock-out)</span></span>
<span id="cb299-5"><a href="mass-spectrometry.html#cb299-5" tabindex="-1"></a>  <span class="fu">impute</span>(base<span class="sc">::</span>min)</span>
<span id="cb299-6"><a href="mass-spectrometry.html#cb299-6" tabindex="-1"></a></span>
<span id="cb299-7"><a href="mass-spectrometry.html#cb299-7" tabindex="-1"></a><span class="co"># plot visualizations comparing normalization methods</span></span>
<span id="cb299-8"><a href="mass-spectrometry.html#cb299-8" tabindex="-1"></a>p03 <span class="ot">&lt;-</span> data_proteins <span class="sc">%&gt;%</span> <span class="fu">plot_normalization</span>()</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-246-1.png" width="768" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb300"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb300-1"><a href="mass-spectrometry.html#cb300-1" tabindex="-1"></a>p04 <span class="ot">&lt;-</span> data_proteins <span class="sc">%&gt;%</span> <span class="fu">plot_variation_cv</span>()</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-247-1.png" width="768" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb301"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb301-1"><a href="mass-spectrometry.html#cb301-1" tabindex="-1"></a>p06 <span class="ot">&lt;-</span> data_proteins <span class="sc">%&gt;%</span> <span class="fu">plot_dynamic_range</span>()</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-249-1.png" width="768" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb302"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb302-1"><a href="mass-spectrometry.html#cb302-1" tabindex="-1"></a><span class="co"># plot visualizations of unbiased clustering</span></span>
<span id="cb302-2"><a href="mass-spectrometry.html#cb302-2" tabindex="-1"></a>p07 <span class="ot">&lt;-</span> data_proteins <span class="sc">%&gt;%</span> <span class="fu">plot_heatmap</span>()</span>
<span id="cb302-3"><a href="mass-spectrometry.html#cb302-3" tabindex="-1"></a>p08 <span class="ot">&lt;-</span> data_proteins <span class="sc">%&gt;%</span> <span class="fu">plot_pca</span>()</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-251-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="expression-analysis" class="section level4 unnumbered hasAnchor">
<h4>Expression Analysis<a href="mass-spectrometry.html#expression-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb303"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb303-1"><a href="mass-spectrometry.html#cb303-1" tabindex="-1"></a>data_proteins <span class="ot">&lt;-</span> data_proteins <span class="sc">%&gt;%</span></span>
<span id="cb303-2"><a href="mass-spectrometry.html#cb303-2" tabindex="-1"></a>  <span class="co"># calculate the expression between experiment: ko and control: wt</span></span>
<span id="cb303-3"><a href="mass-spectrometry.html#cb303-3" tabindex="-1"></a>  <span class="fu">expression</span>(kndw<span class="sc">/</span>ctrl) <span class="sc">%&gt;%</span></span>
<span id="cb303-4"><a href="mass-spectrometry.html#cb303-4" tabindex="-1"></a>  <span class="co"># plot the expression analysis</span></span>
<span id="cb303-5"><a href="mass-spectrometry.html#cb303-5" tabindex="-1"></a>  <span class="fu">plot_volcano</span>(kndw<span class="sc">/</span>ctrl, <span class="at">destination =</span> <span class="st">&quot;png&quot;</span>, <span class="at">significance_column =</span> <span class="st">&quot;p_value&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb303-6"><a href="mass-spectrometry.html#cb303-6" tabindex="-1"></a>  <span class="fu">plot_proportion</span>(kndw<span class="sc">/</span>ctrl, <span class="at">destination =</span> <span class="st">&quot;png&quot;</span>)</span></code></pre></div>
<p>Overall, the tidyproteomics package provides a useful set of tools for preprocessing and analyzing proteomics data using the tidy data framework in R. There are several more workable examples in the online documentation.</p>
<hr />

</div>
</div>
<div id="assayr" class="section level3 hasAnchor" number="10.5.3">
<h3><span class="header-section-number">10.5.3</span> AssayR<a href="mass-spectrometry.html#assayr" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><em>Wills, Jimi, Joy Edwards-Hicks, and Andrew J. Finch. “AssayR: a simple mass spectrometry software tool for targeted metabolic and stable isotope tracer analyses.” Analytical chemistry 89.18 (2017): 9616-9619.</em></p>
<ul>
<li><p><a href="https://bioinformaticshome.com/db/tool/AssayR">WEB tutorial</a></p></li>
<li><p><a href="https://gitlab.com/jimiwills/assay.R/-/blob/master/README.md">WEB manual</a></p></li>
</ul>
<p><code>AssayR</code> is an R package that tailors peak detection for each metabolite in high resolution wide-scan liquid chromatography-mass spectrometry data sets. It integrates peak areas for all isotopologues and outputs extracted ion chromatograms, stacked bar charts, and a .csv data file. <code>AssayR</code> provides easy and robust targeted metabolite and stable isotope analyses on wide-scan data sets from high resolution mass spectrometers, and is shown to provide more accurate and robust quantitation than XCMS.</p>
<hr />
</div>
<div id="deqms" class="section level3 hasAnchor" number="10.5.4">
<h3><span class="header-section-number">10.5.4</span> DEqMS<a href="mass-spectrometry.html#deqms" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><em>Zhu, Yafeng, et al. “DEqMS: a method for accurate variance estimation in differential protein expression analysis.” Molecular &amp; Cellular Proteomics 19.6 (2020): 1047-1057.</em></p>
<ul>
<li><p><a href="https://www.bioconductor.org/packages/release/bioc/manuals/DEqMS/man/DEqMS.pdf">PDF manual</a></p></li>
<li><p><a href="https://www.bioconductor.org/packages/release/bioc/html/DEqMS.html">Bioconductor</a></p></li>
</ul>
<p><code>DEqMS</code> is built on top of Limma, which assumes the same prior variance for all genes. However, protein abundance estimates in proteomics depend on the number of peptides/PSMs quantified, with proteins quantified by multiple peptides or PSMs being more accurately measured. To achieve better accuracy, the <code>DEqMS</code> package estimates different prior variances for proteins quantified by different numbers of PSMs/peptides. This package is suitable for analyzing both label-free and labelled proteomics data.</p>
<hr />
</div>
<div id="ms-empire" class="section level3 hasAnchor" number="10.5.5">
<h3><span class="header-section-number">10.5.5</span> MS-EmpiRe<a href="mass-spectrometry.html#ms-empire" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><em>Ammar, Constantin, et al. “MS-EmpiRe Utilizes Peptide-level Noise Distributions for Ultra-sensitive Detection of Differentially Expressed Proteins [S].” Molecular &amp; Cellular Proteomics 18.9 (2019): 1880-1892.</em></p>
<ul>
<li><a href="https://github.com/zimmerlab/MS-EmpiRe">GitHub</a></li>
</ul>
<p><code>MS-EmpiRe</code> is a new method for mass spectrometry based proteomics that explicitly accounts for the noise underlying peptide fold changes. It derives data set-specific, intensity-dependent empirical error fold change distributions, which are used for individual weighing of peptide fold changes to detect differentially expressed proteins (DEPs). <code>MS-EmpiRe</code> doubles the number of correctly identified DEPs compared to state-of-the-art tools and can be applied to any common quantitative proteomics setup.</p>
<hr />
</div>
<div id="msgrob2" class="section level3 hasAnchor" number="10.5.6">
<h3><span class="header-section-number">10.5.6</span> msgrob2<a href="mass-spectrometry.html#msgrob2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><em>Goeminne L, Gevaert K, Clement L (2016). “Peptide-level Robust Ridge Regression Improves Estimation, Sensitivity, and Specificity in Data-dependent Quantitative Label-free Shotgun Proteomics.” Molecular &amp; Cellular Proteomics, 15(2), 657-668.</em></p>
<ul>
<li><p><a href="https://www.bioconductor.org/packages/release/bioc/manuals/msqrob2/man/msqrob2.pdf">PDF manual</a></p></li>
<li><p><a href="https://www.bioconductor.org/packages/release/bioc/html/msqrob2.html">Bioconductor</a></p></li>
</ul>
<p><code>msgrob2</code> is an R package that provides functions to perform robust estimation in linear models with missing data. With the help of the Expectation-Maximization (EM) algorithm, the package estimates the parameters of the linear model and imputes the missing data. Additionally, the package offers robust methods for estimating the covariance matrix, including the Minimum Covariance Determinant (MCD) estimator and the S-estimator.</p>
<p>The <code>msgrob2</code> package is particularly useful in situations where data is missing from a linear model. The EM algorithm implemented in the package is a powerful tool for imputing missing data, and the robust covariance estimators allow for a better understanding of the data. The package is designed to provide efficient and accurate results when working with incomplete data, making it an essential tool for researchers and data analysts.</p>
<hr />
</div>
<div id="statspro" class="section level3 hasAnchor" number="10.5.7">
<h3><span class="header-section-number">10.5.7</span> StatsPro<a href="mass-spectrometry.html#statspro" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><em>Yang, Yin, et al. “StatsPro: Systematic integration and evaluation of statistical approaches for detecting differential expression in label-free quantitative proteomics.” Journal of Proteomics 250 (2022): 104386.</em></p>
<ul>
<li><p><a href="https://github.com/YanglabWCH/StatsPro#Readme">WEB tutorial</a></p></li>
<li><p><a href="https://github.com/YanglabWCH/StatsPro/blob/main/StatsPro_Manual.pdf">PDF manual</a></p></li>
<li><p><a href="https://github.com/YanglabWCH/StatsPro">GitHub</a></p></li>
</ul>
<p>Quantitative label-free mass spectrometry is a powerful technology for profiling proteins, but choosing an appropriate statistical procedure for detecting differentially expressed proteins remains a challenge. This study presents 12 common testing algorithms and 6 P-value combination methods, along with a user-friendly web tool called StatsPro to help proteomics scientists investigate their influence on Differentially Expressed Protein detection. The authors utilize two case studies demonstrate the tool’s practicability.</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="sharing.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="answers-to-exercises.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/jeffsocal/asms_r_intro/edit/BRANCH/10_01_mass_spectrometry.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": [["ASMS_R_Basics.zip", "HTML"]],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
