<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 10 Mass Spectrometry | Learning R for Mass Spectrometry</title>
  <meta name="description" content="An introductory approach to using R in the analysis of mass spectrometry related data." />
  <meta name="generator" content="bookdown 0.33 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 10 Mass Spectrometry | Learning R for Mass Spectrometry" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="An introductory approach to using R in the analysis of mass spectrometry related data." />
  <meta name="github-repo" content="jeffsocal/ASMS_R_Basics" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 10 Mass Spectrometry | Learning R for Mass Spectrometry" />
  
  <meta name="twitter:description" content="An introductory approach to using R in the analysis of mass spectrometry related data." />
  

<meta name="author" content="Jeff Jones, Heath Patterson, Ryan Benz" />


<meta name="date" content="2023-05-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="sharing.html"/>
<link rel="next" href="answers-to-exercises.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="libs/plotly-binding-4.10.1/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-2.11.1/plotly-latest.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Learning R for Mass Spec.</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="" data-path="acknowledgements.html"><a href="acknowledgements.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="" data-path="contents.html"><a href="contents.html"><i class="fa fa-check"></i>Contents</a></li>
<li class="chapter" data-level="1" data-path="introduction-to-r.html"><a href="introduction-to-r.html"><i class="fa fa-check"></i><b>1</b> Introduction to R</a>
<ul>
<li class="chapter" data-level="" data-path="introduction-to-r.html"><a href="introduction-to-r.html#why-choose-it"><i class="fa fa-check"></i>Why choose it?</a></li>
<li class="chapter" data-level="" data-path="introduction-to-r.html"><a href="introduction-to-r.html#what-you-can-do-with-it"><i class="fa fa-check"></i>What you can do with it?</a></li>
<li class="chapter" data-level="" data-path="introduction-to-r.html"><a href="introduction-to-r.html#the-r-learning-curve"><i class="fa fa-check"></i>The R Learning Curve</a></li>
<li class="chapter" data-level="" data-path="introduction-to-r.html"><a href="introduction-to-r.html#thoughts-about-learning-r-and-how-to-code"><i class="fa fa-check"></i>Thoughts about learning R and how to code</a></li>
<li class="chapter" data-level="" data-path="introduction-to-r.html"><a href="introduction-to-r.html#alternatives"><i class="fa fa-check"></i>Alternatives</a></li>
<li class="chapter" data-level="" data-path="introduction-to-r.html"><a href="introduction-to-r.html#resources"><i class="fa fa-check"></i>Resources</a>
<ul>
<li class="chapter" data-level="" data-path="introduction-to-r.html"><a href="introduction-to-r.html#online"><i class="fa fa-check"></i>Online</a></li>
<li class="chapter" data-level="" data-path="introduction-to-r.html"><a href="introduction-to-r.html#in-print"><i class="fa fa-check"></i>In Print</a></li>
<li class="chapter" data-level="" data-path="introduction-to-r.html"><a href="introduction-to-r.html#organizations"><i class="fa fa-check"></i>Organizations</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="installation.html"><a href="installation.html"><i class="fa fa-check"></i><b>2</b> Installation</a>
<ul>
<li class="chapter" data-level="2.1" data-path="installation.html"><a href="installation.html#r-interpreter"><i class="fa fa-check"></i><b>2.1</b> R interpreter</a></li>
<li class="chapter" data-level="2.2" data-path="installation.html"><a href="installation.html#rstudio"><i class="fa fa-check"></i><b>2.2</b> Rstudio</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="installation.html"><a href="installation.html#ide-layout"><i class="fa fa-check"></i><b>2.2.1</b> IDE Layout</a></li>
<li class="chapter" data-level="2.2.2" data-path="installation.html"><a href="installation.html#usage-..-running-lines-of-code-in-rstudio"><i class="fa fa-check"></i><b>2.2.2</b> Usage .. Running lines of code in RStudio</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="installation.html"><a href="installation.html#packages"><i class="fa fa-check"></i><b>2.3</b> Packages</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="installation.html"><a href="installation.html#what-are-r-packages"><i class="fa fa-check"></i><b>2.3.1</b> What are R Packages?</a></li>
<li class="chapter" data-level="2.3.2" data-path="installation.html"><a href="installation.html#where-to-get-r-packages"><i class="fa fa-check"></i><b>2.3.2</b> Where to get R Packages</a></li>
<li class="chapter" data-level="2.3.3" data-path="installation.html"><a href="installation.html#installing-r-packages"><i class="fa fa-check"></i><b>2.3.3</b> Installing R Packages</a></li>
<li class="chapter" data-level="2.3.4" data-path="installation.html"><a href="installation.html#potential-gotchas"><i class="fa fa-check"></i><b>2.3.4</b> Potential Gotchas</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="installation.html"><a href="installation.html#packages-utilized-in-this-book"><i class="fa fa-check"></i><b>2.4</b> Packages Utilized in This Book</a>
<ul>
<li class="chapter" data-level="" data-path="installation.html"><a href="installation.html#tidyverse"><i class="fa fa-check"></i>tidyverse</a></li>
<li class="chapter" data-level="" data-path="installation.html"><a href="installation.html#mass-spectrometry-specific-packages"><i class="fa fa-check"></i>Mass Spectrometry Specific Packages</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="r-studio-projects.html"><a href="r-studio-projects.html"><i class="fa fa-check"></i><b>3</b> R Studio Projects</a>
<ul>
<li class="chapter" data-level="3.1" data-path="r-studio-projects.html"><a href="r-studio-projects.html#creating"><i class="fa fa-check"></i><b>3.1</b> Creating</a></li>
<li class="chapter" data-level="3.2" data-path="r-studio-projects.html"><a href="r-studio-projects.html#organizing"><i class="fa fa-check"></i><b>3.2</b> Organizing</a>
<ul>
<li class="chapter" data-level="" data-path="r-studio-projects.html"><a href="r-studio-projects.html#data"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="r-studio-projects.html"><a href="r-studio-projects.html#scripts"><i class="fa fa-check"></i>Scripts</a></li>
<li class="chapter" data-level="" data-path="r-studio-projects.html"><a href="r-studio-projects.html#results"><i class="fa fa-check"></i>Results</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-studio-projects.html"><a href="r-studio-projects.html#exercises"><i class="fa fa-check"></i>Exercises</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="r-syntax.html"><a href="r-syntax.html"><i class="fa fa-check"></i><b>4</b> R Syntax</a>
<ul>
<li class="chapter" data-level="4.1" data-path="r-syntax.html"><a href="r-syntax.html#reserved-words"><i class="fa fa-check"></i><b>4.1</b> Reserved Words</a></li>
<li class="chapter" data-level="4.2" data-path="r-syntax.html"><a href="r-syntax.html#types"><i class="fa fa-check"></i><b>4.2</b> Types</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="r-syntax.html"><a href="r-syntax.html#comments"><i class="fa fa-check"></i><b>4.2.1</b> Comments</a></li>
<li class="chapter" data-level="4.2.2" data-path="r-syntax.html"><a href="r-syntax.html#strings"><i class="fa fa-check"></i><b>4.2.2</b> Strings</a></li>
<li class="chapter" data-level="4.2.3" data-path="r-syntax.html"><a href="r-syntax.html#numbers"><i class="fa fa-check"></i><b>4.2.3</b> Numbers</a></li>
<li class="chapter" data-level="4.2.4" data-path="r-syntax.html"><a href="r-syntax.html#operators"><i class="fa fa-check"></i><b>4.2.4</b> Operators</a></li>
<li class="chapter" data-level="4.2.5" data-path="r-syntax.html"><a href="r-syntax.html#variables"><i class="fa fa-check"></i><b>4.2.5</b> Variables</a></li>
<li class="chapter" data-level="4.2.6" data-path="r-syntax.html"><a href="r-syntax.html#statements"><i class="fa fa-check"></i><b>4.2.6</b> Statements</a></li>
<li class="chapter" data-level="4.2.7" data-path="r-syntax.html"><a href="r-syntax.html#functions"><i class="fa fa-check"></i><b>4.2.7</b> Functions</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="r-syntax.html"><a href="r-syntax.html#flow-control"><i class="fa fa-check"></i><b>4.3</b> Flow-Control</a>
<ul>
<li class="chapter" data-level="" data-path="r-syntax.html"><a href="r-syntax.html#for-loop"><i class="fa fa-check"></i>For Loop</a></li>
<li class="chapter" data-level="" data-path="r-syntax.html"><a href="r-syntax.html#if-else"><i class="fa fa-check"></i>If-Else</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-syntax.html"><a href="r-syntax.html#exercises-1"><i class="fa fa-check"></i>Exercises</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="r-objects.html"><a href="r-objects.html"><i class="fa fa-check"></i><b>5</b> R Objects</a>
<ul>
<li class="chapter" data-level="5.1" data-path="r-objects.html"><a href="r-objects.html#variable"><i class="fa fa-check"></i><b>5.1</b> Variable</a></li>
<li class="chapter" data-level="5.2" data-path="r-objects.html"><a href="r-objects.html#vector"><i class="fa fa-check"></i><b>5.2</b> Vector</a></li>
<li class="chapter" data-level="5.3" data-path="r-objects.html"><a href="r-objects.html#matrix"><i class="fa fa-check"></i><b>5.3</b> Matrix</a></li>
<li class="chapter" data-level="5.4" data-path="r-objects.html"><a href="r-objects.html#data-frame"><i class="fa fa-check"></i><b>5.4</b> Data Frame</a></li>
<li class="chapter" data-level="5.5" data-path="r-objects.html"><a href="r-objects.html#list"><i class="fa fa-check"></i><b>5.5</b> List</a></li>
<li class="chapter" data-level="5.6" data-path="r-objects.html"><a href="r-objects.html#tibbles"><i class="fa fa-check"></i><b>5.6</b> Tibbles</a></li>
<li class="chapter" data-level="" data-path="r-objects.html"><a href="r-objects.html#how-to-tell-what-you-are-dealing-with"><i class="fa fa-check"></i>How to tell what you are dealing with</a></li>
<li class="chapter" data-level="" data-path="r-objects.html"><a href="r-objects.html#exercises-2"><i class="fa fa-check"></i>Exercises</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="tidyverse-1.html"><a href="tidyverse-1.html"><i class="fa fa-check"></i><b>6</b> Tidyverse</a>
<ul>
<li class="chapter" data-level="6.1" data-path="tidyverse-1.html"><a href="tidyverse-1.html#core-packages"><i class="fa fa-check"></i><b>6.1</b> Core Packages</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="tidyverse-1.html"><a href="tidyverse-1.html#magrittr"><i class="fa fa-check"></i><b>6.1.1</b> magrittr</a></li>
<li class="chapter" data-level="6.1.2" data-path="tidyverse-1.html"><a href="tidyverse-1.html#tibble"><i class="fa fa-check"></i><b>6.1.2</b> tibble</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="tidyverse-1.html"><a href="tidyverse-1.html#importing"><i class="fa fa-check"></i><b>6.2</b> Importing</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="tidyverse-1.html"><a href="tidyverse-1.html#readr"><i class="fa fa-check"></i><b>6.2.1</b> readr</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="tidyverse-1.html"><a href="tidyverse-1.html#wrangling"><i class="fa fa-check"></i><b>6.3</b> Wrangling</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="tidyverse-1.html"><a href="tidyverse-1.html#dplyr"><i class="fa fa-check"></i><b>6.3.1</b> dplyr</a></li>
<li class="chapter" data-level="6.3.2" data-path="tidyverse-1.html"><a href="tidyverse-1.html#tidyr"><i class="fa fa-check"></i><b>6.3.2</b> tidyr</a></li>
<li class="chapter" data-level="6.3.3" data-path="tidyverse-1.html"><a href="tidyverse-1.html#purrr"><i class="fa fa-check"></i><b>6.3.3</b> purrr</a></li>
<li class="chapter" data-level="6.3.4" data-path="tidyverse-1.html"><a href="tidyverse-1.html#glue"><i class="fa fa-check"></i><b>6.3.4</b> glue</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="tidyverse-1.html"><a href="tidyverse-1.html#data-types"><i class="fa fa-check"></i><b>6.4</b> Data Types</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="tidyverse-1.html"><a href="tidyverse-1.html#stringr"><i class="fa fa-check"></i><b>6.4.1</b> stringr</a></li>
<li class="chapter" data-level="6.4.2" data-path="tidyverse-1.html"><a href="tidyverse-1.html#lubridate"><i class="fa fa-check"></i><b>6.4.2</b> lubridate</a></li>
<li class="chapter" data-level="6.4.3" data-path="tidyverse-1.html"><a href="tidyverse-1.html#forcats"><i class="fa fa-check"></i><b>6.4.3</b> forcats</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="tidyverse-1.html"><a href="tidyverse-1.html#summarizing"><i class="fa fa-check"></i><b>6.5</b> Summarizing</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="tidyverse-1.html"><a href="tidyverse-1.html#dplyr-1"><i class="fa fa-check"></i><b>6.5.1</b> dplyr</a></li>
<li class="chapter" data-level="6.5.2" data-path="tidyverse-1.html"><a href="tidyverse-1.html#ggplot2"><i class="fa fa-check"></i><b>6.5.2</b> ggplot2</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tidyverse-1.html"><a href="tidyverse-1.html#exercises-3"><i class="fa fa-check"></i>Exercises</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="data-wrangling.html"><a href="data-wrangling.html"><i class="fa fa-check"></i><b>7</b> Data Wrangling</a>
<ul>
<li class="chapter" data-level="7.1" data-path="data-wrangling.html"><a href="data-wrangling.html#tidy-data"><i class="fa fa-check"></i><b>7.1</b> Tidy Data</a></li>
<li class="chapter" data-level="7.2" data-path="data-wrangling.html"><a href="data-wrangling.html#example-walkthrough"><i class="fa fa-check"></i><b>7.2</b> Example Walkthrough</a>
<ul>
<li class="chapter" data-level="" data-path="data-wrangling.html"><a href="data-wrangling.html#read_"><i class="fa fa-check"></i>read_</a></li>
<li class="chapter" data-level="" data-path="data-wrangling.html"><a href="data-wrangling.html#separate"><i class="fa fa-check"></i>separate</a></li>
<li class="chapter" data-level="" data-path="data-wrangling.html"><a href="data-wrangling.html#mutate"><i class="fa fa-check"></i>mutate</a></li>
<li class="chapter" data-level="" data-path="data-wrangling.html"><a href="data-wrangling.html#pivot_"><i class="fa fa-check"></i>pivot_</a></li>
<li class="chapter" data-level="" data-path="data-wrangling.html"><a href="data-wrangling.html#write_"><i class="fa fa-check"></i>write_</a></li>
<li class="chapter" data-level="" data-path="data-wrangling.html"><a href="data-wrangling.html#non-pipe"><i class="fa fa-check"></i>Non-pipe</a></li>
<li class="chapter" data-level="" data-path="data-wrangling.html"><a href="data-wrangling.html#summarize"><i class="fa fa-check"></i>summarize</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="data-wrangling.html"><a href="data-wrangling.html#exercises---evalfalse"><i class="fa fa-check"></i><b>7.3</b> Exercises {-, eval=FALSE}</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="data-visualization.html"><a href="data-visualization.html"><i class="fa fa-check"></i><b>8</b> Data Visualization</a>
<ul>
<li class="chapter" data-level="8.1" data-path="data-visualization.html"><a href="data-visualization.html#base-plot"><i class="fa fa-check"></i><b>8.1</b> Base: plot</a></li>
<li class="chapter" data-level="8.2" data-path="data-visualization.html"><a href="data-visualization.html#ggplot2-1"><i class="fa fa-check"></i><b>8.2</b> GGPlot2</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="data-visualization.html"><a href="data-visualization.html#syntax"><i class="fa fa-check"></i><b>8.2.1</b> Syntax</a></li>
<li class="chapter" data-level="8.2.2" data-path="data-visualization.html"><a href="data-visualization.html#basic-data-plotting"><i class="fa fa-check"></i><b>8.2.2</b> Basic Data Plotting</a></li>
<li class="chapter" data-level="8.2.3" data-path="data-visualization.html"><a href="data-visualization.html#data-distributions"><i class="fa fa-check"></i><b>8.2.3</b> Data Distributions</a></li>
<li class="chapter" data-level="8.2.4" data-path="data-visualization.html"><a href="data-visualization.html#extended-syntax"><i class="fa fa-check"></i><b>8.2.4</b> Extended Syntax</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="data-visualization.html"><a href="data-visualization.html#alternatives-1"><i class="fa fa-check"></i><b>8.3</b> Alternatives</a>
<ul>
<li class="chapter" data-level="" data-path="data-visualization.html"><a href="data-visualization.html#lattice"><i class="fa fa-check"></i>lattice</a></li>
<li class="chapter" data-level="" data-path="data-visualization.html"><a href="data-visualization.html#plotly"><i class="fa fa-check"></i>plotly</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-visualization.html"><a href="data-visualization.html#exercises-4"><i class="fa fa-check"></i>Exercises</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="sharing.html"><a href="sharing.html"><i class="fa fa-check"></i><b>9</b> Sharing</a>
<ul>
<li class="chapter" data-level="9.1" data-path="sharing.html"><a href="sharing.html#notebooks"><i class="fa fa-check"></i><b>9.1</b> Notebooks</a>
<ul>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#using-notebooks"><i class="fa fa-check"></i>Using Notebooks</a></li>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#sharing-a-notebook"><i class="fa fa-check"></i>Sharing A Notebook</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="sharing.html"><a href="sharing.html#packrat"><i class="fa fa-check"></i><b>9.2</b> Packrat</a>
<ul>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#initiating-a-packrat-project"><i class="fa fa-check"></i>Initiating a Packrat Project</a></li>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#installing-a-package-into-the-project-library"><i class="fa fa-check"></i>Installing a Package into the Project Library</a></li>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#loading-a-package-from-the-project-library"><i class="fa fa-check"></i>Loading a Package from the Project Library</a></li>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#updating-a-package-in-the-project-library"><i class="fa fa-check"></i>Updating a Package in the Project Library</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="sharing.html"><a href="sharing.html#github"><i class="fa fa-check"></i><b>9.3</b> GitHub</a>
<ul>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#how-to-use-github"><i class="fa fa-check"></i>How to Use GitHub</a></li>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#how-to-share-a-github-repository"><i class="fa fa-check"></i>How to Share A GitHub Repository</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="sharing.html"><a href="sharing.html#docker"><i class="fa fa-check"></i><b>9.4</b> Docker</a></li>
<li class="chapter" data-level="9.5" data-path="sharing.html"><a href="sharing.html#r-packages"><i class="fa fa-check"></i><b>9.5</b> R Packages</a>
<ul>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#creating-a-package-in-rstudio"><i class="fa fa-check"></i>Creating a Package in RStudio</a></li>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#using-the-package"><i class="fa fa-check"></i>Using the Package</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="sharing.html"><a href="sharing.html#r-shiny-applications"><i class="fa fa-check"></i><b>9.6</b> R Shiny Applications</a>
<ul>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#how-to-create-an-r-shiny-application-in-the-rstudio-ide"><i class="fa fa-check"></i>How to Create an R Shiny Application in the RStudio IDE</a></li>
<li class="chapter" data-level="" data-path="sharing.html"><a href="sharing.html#example-of-an-r-shiny-application-for-plotting-points"><i class="fa fa-check"></i>Example of an R Shiny Application for Plotting Points</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html"><i class="fa fa-check"></i><b>10</b> Mass Spectrometry</a>
<ul>
<li class="chapter" data-level="10.1" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#data-formats"><i class="fa fa-check"></i><b>10.1</b> Data Formats</a></li>
<li class="chapter" data-level="10.2" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#commercial"><i class="fa fa-check"></i><b>10.2</b> Commercial</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#raw-thermo"><i class="fa fa-check"></i><b>10.2.1</b> RAW (Thermo)</a></li>
<li class="chapter" data-level="10.2.2" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#wiff-sciex"><i class="fa fa-check"></i><b>10.2.2</b> WIFF (Sciex)</a></li>
<li class="chapter" data-level="10.2.3" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#d-agilent"><i class="fa fa-check"></i><b>10.2.3</b> D (Agilent)</a></li>
<li class="chapter" data-level="10.2.4" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#d-bruker-timstof-data"><i class="fa fa-check"></i><b>10.2.4</b> .d (Bruker) timsTOF data</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#open-access-data"><i class="fa fa-check"></i><b>10.3</b> Open Access Data</a>
<ul>
<li class="chapter" data-level="" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#mzxml"><i class="fa fa-check"></i>mzXML</a></li>
<li class="chapter" data-level="" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#mzml"><i class="fa fa-check"></i>mzML</a></li>
<li class="chapter" data-level="" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#mzmlb"><i class="fa fa-check"></i>mzMLb</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#r-calculate-mass"><i class="fa fa-check"></i><b>10.4</b> R Calculate Mass</a>
<ul>
<li class="chapter" data-level="" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#brain"><i class="fa fa-check"></i>BRAIN</a></li>
<li class="chapter" data-level="10.4.1" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#envipat"><i class="fa fa-check"></i><b>10.4.1</b> enviPat</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#r-ms-data"><i class="fa fa-check"></i><b>10.5</b> R MS Data</a>
<ul>
<li class="chapter" data-level="" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#xcms"><i class="fa fa-check"></i>xcms</a></li>
<li class="chapter" data-level="" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#mzr"><i class="fa fa-check"></i>mzR</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#r-ms-analysis"><i class="fa fa-check"></i><b>10.6</b> R MS Analysis</a>
<ul>
<li class="chapter" data-level="" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#deqms"><i class="fa fa-check"></i>deqms</a></li>
<li class="chapter" data-level="" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#msempire"><i class="fa fa-check"></i>msempire</a></li>
<li class="chapter" data-level="" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#msqrob"><i class="fa fa-check"></i>msqrob</a></li>
<li class="chapter" data-level="" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#msstats"><i class="fa fa-check"></i>MSstats</a></li>
<li class="chapter" data-level="" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#statspro-r-package"><i class="fa fa-check"></i>StatsPro R Package</a></li>
<li class="chapter" data-level="" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#msgrob2"><i class="fa fa-check"></i>msgrob2</a></li>
<li class="chapter" data-level="" data-path="mass-spectrometry.html"><a href="mass-spectrometry.html#tidyproteomics"><i class="fa fa-check"></i>Tidyproteomics</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="answers-to-exercises.html"><a href="answers-to-exercises.html"><i class="fa fa-check"></i>Answers to Exercises</a>
<ul>
<li class="chapter" data-level="" data-path="answers-to-exercises.html"><a href="answers-to-exercises.html#r-syntax-1"><i class="fa fa-check"></i><strong>4</strong> R Syntax</a></li>
<li class="chapter" data-level="" data-path="answers-to-exercises.html"><a href="answers-to-exercises.html#r-objects-1"><i class="fa fa-check"></i><strong>5</strong> R Objects</a></li>
<li class="chapter" data-level="" data-path="answers-to-exercises.html"><a href="answers-to-exercises.html#tidyverse-2"><i class="fa fa-check"></i><strong>6</strong> Tidyverse</a></li>
<li class="chapter" data-level="" data-path="answers-to-exercises.html"><a href="answers-to-exercises.html#data-wrangling-1"><i class="fa fa-check"></i><strong>7</strong> Data Wrangling</a></li>
<li class="chapter" data-level="" data-path="answers-to-exercises.html"><a href="answers-to-exercises.html#data-wrangling-2"><i class="fa fa-check"></i><strong>7</strong> Data Wrangling</a></li>
<li class="chapter" data-level="" data-path="answers-to-exercises.html"><a href="answers-to-exercises.html#data-visualization-1"><i class="fa fa-check"></i><strong>8</strong> Data Visualization</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/jeffsocal/asms_r_intro" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Learning R for Mass Spectrometry</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="mass-spectrometry" class="section level1 hasAnchor" number="10">
<h1><span class="header-section-number">Chapter 10</span> Mass Spectrometry<a href="mass-spectrometry.html#mass-spectrometry" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><br />
</p>
<table>
<colgroup>
<col width="32%" />
<col width="67%" />
</colgroup>
<tbody>
<tr class="odd">
<td><img src="images/03.png" height="68" /></td>
<td><p><strong>At the end of this chapter you should be able to</strong></p>
<ul>
<li><p>Understand commercially proprietary raw data.</p></li>
<li><p>How to convert raw data to an open access format.</p></li>
<li><p>Be familiar with some R packages specific to mass spectrometry.</p></li>
</ul></td>
</tr>
</tbody>
</table>
<p><br />
</p>

<div id="data-formats" class="section level2 hasAnchor" number="10.1">
<h2><span class="header-section-number">10.1</span> Data Formats<a href="mass-spectrometry.html#data-formats" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Accessing mass spectrometry data has some barriers due to vendors proprietary file formats. In recent years, vendors have been providing software development kits (SDKs) that allow programmers to provide direct access to the raw data. This is essential when developing fundamental workflows that require such data and also can enable analysis tools to bypass sometimes time consuming file format conversions to open formats. There are some means to access such raw data through R, allowing end-to-end pipelines to be developed directly from vendor data.</p>
</div>
<div id="commercial" class="section level2 hasAnchor" number="10.2">
<h2><span class="header-section-number">10.2</span> Commercial<a href="mass-spectrometry.html#commercial" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="raw-thermo" class="section level3 hasAnchor" number="10.2.1">
<h3><span class="header-section-number">10.2.1</span> RAW (Thermo)<a href="mass-spectrometry.html#raw-thermo" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="float">
<img src="https://github.com/fgcz/rawrr/raw/devel/rawrr_logo.png" alt="rawrr R package" />
<div class="figcaption">rawrr R package</div>
</div>
<p><strong>rawrr</strong> : <a href="https://github.com/fgcz/rawrr" class="uri">https://github.com/fgcz/rawrr</a>
The package provides access to proprietary Thermo Fisher Scientific Orbitrap instrument data as a stand-alone R package or serves as MsRawFileReaderBackend for the Bioconductor Spectra package. rawrr wraps the functionality of the RawFileReader .NET assembly. Test files are provided by the tartare ExperimentData package.</p>
<p><strong>rawReadeR</strong> : <a href="https://github.com/wilsontom/rawReadeR" class="uri">https://github.com/wilsontom/rawReadeR</a>
rawReadeR is an R/C++ API to the MSFileReader.dll. rawReadeR allows for m/z profiles and scan header data to be extracted from .RAW files without the need for manual extraction using Xcalibur or conversion to a more universal format (.mzML, .mZXML, etc…)</p>
</div>
<div id="wiff-sciex" class="section level3 hasAnchor" number="10.2.2">
<h3><span class="header-section-number">10.2.2</span> WIFF (Sciex)<a href="mass-spectrometry.html#wiff-sciex" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>None known at the moment.</p>
</div>
<div id="d-agilent" class="section level3 hasAnchor" number="10.2.3">
<h3><span class="header-section-number">10.2.3</span> D (Agilent)<a href="mass-spectrometry.html#d-agilent" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Convert from raw using vendor software.</p>
</div>
<div id="d-bruker-timstof-data" class="section level3 hasAnchor" number="10.2.4">
<h3><span class="header-section-number">10.2.4</span> .d (Bruker) timsTOF data<a href="mass-spectrometry.html#d-bruker-timstof-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Bruker has released an SDK that has now been ported to python and R.</p>
<p><em>timsR</em> : <a href="https://github.com/MatteoLacki/timsr" class="uri">https://github.com/MatteoLacki/timsr</a></p>

</div>
</div>
<div id="open-access-data" class="section level2 hasAnchor" number="10.3">
<h2><span class="header-section-number">10.3</span> Open Access Data<a href="mass-spectrometry.html#open-access-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Mass spectrometry (MS) is a powerful analytical method that can be used to determine the mass-to-charge ratio (m/z) of ions in a sample. Mass spectrometry data is generated from MS experiments and can be stored in various formats, including mzXML, mzML, and mzH5.</p>
<p><em>ProteoWizard</em> is an open-source software suite that provides a collection of open-source, cross-platform software libraries and tools for extracting raw mass spectrometry data from from various instrument vendor formats and converting it to the formats listed below.</p>
<blockquote>
<p>_Kessner, D., Chambers, M., Burke, R., Agus, D., &amp; Mallick, P. (2008). ProteoWizard: open source software for rapid proteomics tools development. Bioinformatics, 24(21), 2534–2536.</p>
<p><a href="https://proteowizard.sourceforge.io/">proteowizard.sourceforge.io/</a></p>
</blockquote>
<div id="mzxml" class="section level3 unnumbered hasAnchor">
<h3>mzXML<a href="mass-spectrometry.html#mzxml" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>mzXML is an open XML-based format for encoding MS data. It was developed by the Seattle Proteome Center and is widely used in the mass spectrometry community. mzXML files contain raw MS data, as well as metadata describing the instrument parameters used to acquire the data. mzXML files can be processed using a variety of software tools, such as the Trans-Proteomic Pipeline and ProteoWizard.</p>
<p>The mzXML format has been shown to be effective in handling data from a wide range of instruments. It has a simple structure that makes it easy to parse and process, making it an attractive choice for many researchers. The format is also relatively lightweight, which makes it easy to transfer and store large amounts of data.</p>
<blockquote>
<p><em>Pedrioli, P.G., Eng, J.K., Hubley, R., Vogelzang, M., Deutsch, E.W., Raught, B., Pratt, B., Nilsson, E., Angeletti, R.H., Apweiler, R. and Cheung, K., 2004. A common open representation of mass spectrometry data and its application to proteomics research. Nature biotechnology, 22(11), pp.1459-1466.</em></p>
</blockquote>
</div>
<div id="mzml" class="section level3 unnumbered hasAnchor">
<h3>mzML<a href="mass-spectrometry.html#mzml" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The mzML format is another open XML-based format for MS data, developed by the Proteomics Standards Initiative. It is designed to be more flexible than mzXML and includes more detailed metadata. mzML files can be processed using software tools such as OpenMS and mzR.</p>
<p>The mzML format allows for more detailed and comprehensive data storage than mzXML. This is because mzML has a more complex structure, which enables the storage of a wider range of experimental metadata. The format is also more flexible, which means that it can be easily adapted to different types of experiments and instruments.</p>
<blockquote>
<p><em>Martens, L., Chambers, M., Sturm, M., Kessner, D., Levander, F., Shofstahl, J., Tang, W.H., Römpp, A., Neumann, S., Pizarro, A.D. and Montecchi-Palazzi, L., 2011. mzML—a community standard for mass spectrometry data. Molecular &amp; cellular proteomics, 10(1), p.R110. 000133.</em></p>
</blockquote>
</div>
<div id="mzmlb" class="section level3 unnumbered hasAnchor">
<h3>mzMLb<a href="mass-spectrometry.html#mzmlb" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Recently proposed as a new file format based on HDF5 and NetCDF4 standards, mzMLb is faster and more flexible than existing approaches while preserving the XML encoding of metadata. Additionally, it is optimized for both read/write speed and storage efficiency. The format has a reference implementation provided within the ProteoWizard toolkit.</p>
<blockquote>
<p><em>Bhamber, Ranjeet S., et al. “mzMLb: A future-proof raw mass spectrometry data format based on standards-compliant mzML and optimized for speed and storage requirements.” Journal of proteome research 20.1 (2020): 172-183.</em></p>
</blockquote>

</div>
</div>
<div id="r-calculate-mass" class="section level2 hasAnchor" number="10.4">
<h2><span class="header-section-number">10.4</span> R Calculate Mass<a href="mass-spectrometry.html#r-calculate-mass" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Molecular mass is the sum of the atomic masses of all the atoms in a molecule. It is an important parameter used in various fields of chemistry. The molecular mass of a molecule is usually expressed in atomic mass units (amu) or daltons (Da). Mass spectrometry based measurements require a charge and are expressed as mass-to-charge (m/z) or as Thompsons (Th).</p>
<p>Isotopic probabilities are also important in determining molecular mass. Isotopes are atoms of the same element that have different numbers of neutrons in their nuclei. Isotopes of an element have different atomic masses. The isotopic probability of an element is the probability that a given isotope of that element will occur in nature. For example, carbon (C) has two stable isotopes, carbon-12 (12C) and carbon-13 (13C), with atomic masses of 12.000 amu and 13.003 amu, respectively. The isotopic probability of carbon-12 is 98.9%, while that of carbon-13 is 1.1%. Therefore, the average atomic mass of carbon in nature is 12.011 amu.</p>
<p>There are several measures of a molecular mass that can be reported. The <em>average mass</em> is that of the weighted average of all isotopes and likely to be reported for small molecules. While the <em>nominal mass</em> of a molecule is defined as the sum of the integer masses of the most abundant isotopes in a molecule. The <em>monoisotopic mass</em> is commonly considered the as the sum of the exact masses of the lightest isotopes, and this value is considered in all peptide-based proteomics applications.</p>
<div id="brain" class="section level3 unnumbered hasAnchor">
<h3>BRAIN<a href="mass-spectrometry.html#brain" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Baffling Recursive Algorithm for Isotope distributioN calculations</strong></p>
<p>R BRAIN is an isotopic abundance calculator implemented in R programming language and is especially useful for chemists and researchers who deal with complex molecules and need to calculate their isotopic composition accurately. In addition, it has a handy function for calculating the mass directly from an amino acid sequence.</p>
<p><br />
</p>
<table>
<colgroup>
<col width="23%" />
<col width="76%" />
</colgroup>
<tbody>
<tr class="odd">
<td><img src="images/05.png" height="68" /></td>
<td><p><strong>Documentation</strong> | <a href="https://bioconductor.org/packages/release/bioc/manuals/BRAIN/man/BRAIN.pdf">Bioconductor</a></p>
<p><strong>Literature</strong> | <em>Analytical chemistry</em> 2013 85(4), 1991-1994</p></td>
</tr>
</tbody>
</table>
<p><br />
</p>
<div id="installation-1" class="section level4 unnumbered hasAnchor">
<h4>Installation<a href="mass-spectrometry.html#installation-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb228"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb228-1"><a href="mass-spectrometry.html#cb228-1" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;BRAIN&quot;</span>)</span></code></pre></div>
</div>
<div id="use" class="section level4 unnumbered hasAnchor">
<h4>Use<a href="mass-spectrometry.html#use" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb229"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb229-1"><a href="mass-spectrometry.html#cb229-1" tabindex="-1"></a><span class="fu">library</span>(BRAIN)</span>
<span id="cb229-2"><a href="mass-spectrometry.html#cb229-2" tabindex="-1"></a></span>
<span id="cb229-3"><a href="mass-spectrometry.html#cb229-3" tabindex="-1"></a><span class="co"># Human insulin amino acid sequence</span></span>
<span id="cb229-4"><a href="mass-spectrometry.html#cb229-4" tabindex="-1"></a>str_seq <span class="ot">&lt;-</span> <span class="st">&quot;MALWMRLLPLLALLALWGPDPAAAFVNQHLCGSHLVEALYLVCGERGFFYTPKTRREAEDLQVGQVELGGGPGAGSLQPLALEGSLQKRGIVEQCCTSICSLYQLENYCN&quot;</span></span>
<span id="cb229-5"><a href="mass-spectrometry.html#cb229-5" tabindex="-1"></a></span>
<span id="cb229-6"><a href="mass-spectrometry.html#cb229-6" tabindex="-1"></a><span class="co"># get a list-object of atoms</span></span>
<span id="cb229-7"><a href="mass-spectrometry.html#cb229-7" tabindex="-1"></a>lst_atm <span class="ot">&lt;-</span> <span class="fu">getAtomsFromSeq</span>(str_seq) </span></code></pre></div>
<p>Calculate the average mass.</p>
<div class="sourceCode" id="cb230"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb230-1"><a href="mass-spectrometry.html#cb230-1" tabindex="-1"></a><span class="fu">calculateAverageMass</span>(lst_atm)</span></code></pre></div>
<pre><code>## [1] 11980.82</code></pre>
<p>Calculate the monoisotopic mass.</p>
<div class="sourceCode" id="cb232"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb232-1"><a href="mass-spectrometry.html#cb232-1" tabindex="-1"></a><span class="fu">calculateMonoisotopicMass</span>(lst_atm)</span></code></pre></div>
<pre><code>## [1] 11973.02</code></pre>
<p>Calculate the isotopic abundances (probabilities) mass.</p>
<div class="sourceCode" id="cb234"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb234-1"><a href="mass-spectrometry.html#cb234-1" tabindex="-1"></a>lst_isotopes <span class="ot">&lt;-</span> <span class="fu">useBRAIN</span>(lst_atm, <span class="at">nrPeaks =</span> <span class="dv">20</span>)</span>
<span id="cb234-2"><a href="mass-spectrometry.html#cb234-2" tabindex="-1"></a></span>
<span id="cb234-3"><a href="mass-spectrometry.html#cb234-3" tabindex="-1"></a><span class="fu">plot</span>(lst_isotopes<span class="sc">$</span>masses, lst_isotopes<span class="sc">$</span>isoDistr, <span class="at">xlab=</span><span class="st">&#39;Mass&#39;</span>, <span class="at">ylab=</span><span class="st">&#39;Probability&#39;</span>, <span class="at">type =</span> <span class="st">&#39;h&#39;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-187-1.png" width="576" style="display: block; margin: auto;" />
Calculate the isotopic abundances (probabilities) mass for a metabolite [C100H200S2Cl5].</p>
<div class="sourceCode" id="cb235"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb235-1"><a href="mass-spectrometry.html#cb235-1" tabindex="-1"></a>lst_atm <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">C=</span><span class="dv">100</span>, <span class="at">H=</span><span class="dv">200</span>, <span class="at">S=</span><span class="dv">2</span>, <span class="at">Cl=</span><span class="dv">5</span>)</span>
<span id="cb235-2"><a href="mass-spectrometry.html#cb235-2" tabindex="-1"></a></span>
<span id="cb235-3"><a href="mass-spectrometry.html#cb235-3" tabindex="-1"></a>lst_isotopes <span class="ot">&lt;-</span> <span class="fu">useBRAIN</span>(lst_atm, <span class="at">nrPeaks =</span> <span class="dv">13</span>)</span>
<span id="cb235-4"><a href="mass-spectrometry.html#cb235-4" tabindex="-1"></a></span>
<span id="cb235-5"><a href="mass-spectrometry.html#cb235-5" tabindex="-1"></a><span class="fu">plot</span>(<span class="at">x =</span> lst_isotopes<span class="sc">$</span>masses, </span>
<span id="cb235-6"><a href="mass-spectrometry.html#cb235-6" tabindex="-1"></a>     <span class="at">y =</span> lst_isotopes<span class="sc">$</span>isoDistr <span class="sc">/</span> <span class="fu">max</span>(lst_isotopes<span class="sc">$</span>isoDistr) <span class="sc">*</span> <span class="dv">100</span>, </span>
<span id="cb235-7"><a href="mass-spectrometry.html#cb235-7" tabindex="-1"></a>     <span class="at">xlab=</span><span class="st">&#39;Mass&#39;</span>, <span class="at">ylab=</span><span class="st">&#39;Relative abundance&#39;</span>, <span class="at">type =</span> <span class="st">&#39;h&#39;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-188-1.png" width="576" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="envipat" class="section level3 hasAnchor" number="10.4.1">
<h3><span class="header-section-number">10.4.1</span> enviPat<a href="mass-spectrometry.html#envipat" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Isotope Pattern, Profile and Centroid Calculation for Mass Spectrometry</strong></p>
<p>A new method (year 2022) for calculating theoretical isotope patterns in mass spectrometry. This method uses a treelike structure to derive sets of subisotopologues for each element in a molecule. By doing so, it allows for early pruning of low-probability isotopologues and the detection of the most probable isotopologue. The method was validated in a large-scale benchmark simulation.</p>
<p><br />
</p>
<table>
<colgroup>
<col width="27%" />
<col width="72%" />
</colgroup>
<tbody>
<tr class="odd">
<td><img src="images/05.png" height="68" /></td>
<td><p><strong>Documentation</strong> | <a href="https://cran.r-project.org/web/packages/enviPat/enviPat.pdf">CRAN</a></p>
<p><strong>Literature</strong> | <em>Anal. Chem.</em> 2015, 87, 11, 5738–5744</p></td>
</tr>
</tbody>
</table>
<p><br />
</p>
<div id="installation-2" class="section level4 unnumbered hasAnchor">
<h4>Installation<a href="mass-spectrometry.html#installation-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb236"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb236-1"><a href="mass-spectrometry.html#cb236-1" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;enviPat&quot;</span>)</span></code></pre></div>
</div>
<div id="use-1" class="section level4 unnumbered hasAnchor">
<h4>Use<a href="mass-spectrometry.html#use-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb237"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb237-1"><a href="mass-spectrometry.html#cb237-1" tabindex="-1"></a><span class="fu">library</span>(enviPat)</span>
<span id="cb237-2"><a href="mass-spectrometry.html#cb237-2" tabindex="-1"></a></span>
<span id="cb237-3"><a href="mass-spectrometry.html#cb237-3" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;isotopes&quot;</span>)</span>
<span id="cb237-4"><a href="mass-spectrometry.html#cb237-4" tabindex="-1"></a></span>
<span id="cb237-5"><a href="mass-spectrometry.html#cb237-5" tabindex="-1"></a>pattern <span class="ot">&lt;-</span> <span class="fu">isopattern</span>(</span>
<span id="cb237-6"><a href="mass-spectrometry.html#cb237-6" tabindex="-1"></a>  isotopes,</span>
<span id="cb237-7"><a href="mass-spectrometry.html#cb237-7" tabindex="-1"></a>  <span class="st">&quot;C100H200S2Cl5&quot;</span>,</span>
<span id="cb237-8"><a href="mass-spectrometry.html#cb237-8" tabindex="-1"></a>  <span class="at">threshold=</span><span class="fl">0.1</span>,</span>
<span id="cb237-9"><a href="mass-spectrometry.html#cb237-9" tabindex="-1"></a>  <span class="at">plotit=</span><span class="cn">TRUE</span>,</span>
<span id="cb237-10"><a href="mass-spectrometry.html#cb237-10" tabindex="-1"></a>  <span class="at">charge=</span><span class="cn">FALSE</span>,</span>
<span id="cb237-11"><a href="mass-spectrometry.html#cb237-11" tabindex="-1"></a>  <span class="at">emass=</span><span class="fl">0.00054858</span>,</span>
<span id="cb237-12"><a href="mass-spectrometry.html#cb237-12" tabindex="-1"></a>  <span class="at">algo=</span><span class="dv">1</span></span>
<span id="cb237-13"><a href="mass-spectrometry.html#cb237-13" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-190-1.png" width="576" style="display: block; margin: auto;" /></p>

</div>
</div>
</div>
<div id="r-ms-data" class="section level2 hasAnchor" number="10.5">
<h2><span class="header-section-number">10.5</span> R MS Data<a href="mass-spectrometry.html#r-ms-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<!--# HEATH -->
<div id="xcms" class="section level3 unnumbered hasAnchor">
<h3>xcms<a href="mass-spectrometry.html#xcms" class="anchor-section" aria-label="Anchor link to header"></a></h3>
</div>
<div id="mzr" class="section level3 unnumbered hasAnchor">
<h3>mzR<a href="mass-spectrometry.html#mzr" class="anchor-section" aria-label="Anchor link to header"></a></h3>

</div>
</div>
<div id="r-ms-analysis" class="section level2 hasAnchor" number="10.6">
<h2><span class="header-section-number">10.6</span> R MS Analysis<a href="mass-spectrometry.html#r-ms-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Finally, there are numerous R packages that implement methods, common and advanced, in statistical post analysis of quantitative data. Explored here are only a few of the available packages.</p>
<div id="deqms" class="section level3 unnumbered hasAnchor">
<h3>deqms<a href="mass-spectrometry.html#deqms" class="anchor-section" aria-label="Anchor link to header"></a></h3>
</div>
<div id="msempire" class="section level3 unnumbered hasAnchor">
<h3>msempire<a href="mass-spectrometry.html#msempire" class="anchor-section" aria-label="Anchor link to header"></a></h3>
</div>
<div id="msqrob" class="section level3 unnumbered hasAnchor">
<h3>msqrob<a href="mass-spectrometry.html#msqrob" class="anchor-section" aria-label="Anchor link to header"></a></h3>

</div>
<div id="msstats" class="section level3 unnumbered hasAnchor">
<h3>MSstats<a href="mass-spectrometry.html#msstats" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The MSstats package is an R package designed for the analysis of label-free mass spectrometry data. It provides a wide range of statistical tools for the analysis of protein abundance data, including normalization, missing value imputation, quality control, and differential expression analysis. MSstats provides a powerful and flexible way to analyze mass spectrometry data, making it an essential tool for researchers in the field.</p>
<p><br />
</p>
<table>
<colgroup>
<col width="20%" />
<col width="80%" />
</colgroup>
<tbody>
<tr class="odd">
<td><img src="images/05.png" height="68" /></td>
<td><p><strong>Documentation</strong> | <a href="https://msstats.org">web</a>, <a href="https://bioconductor.org/packages/release/bioc/manuals/MSstats/man/MSstats.pdf">PDF</a></p>
<p><strong>Code</strong> | <a href="https://www.bioconductor.org/packages/release/bioc/html/MSstats.html">Bioconductor</a></p>
<p><strong>Literature</strong> | <em>Bioinformatics</em> 30.17 (2014): 2524-2526</p></td>
</tr>
</tbody>
</table>
<p><br />
</p>
<div id="installation-3" class="section level4 unnumbered hasAnchor">
<h4>Installation<a href="mass-spectrometry.html#installation-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Install MSstats from Bioconductor:</p>
<div class="sourceCode" id="cb238"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb238-1"><a href="mass-spectrometry.html#cb238-1" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;MSstats&quot;</span>)</span></code></pre></div>
</div>
<div id="loading-data" class="section level4 unnumbered hasAnchor">
<h4>Loading Data<a href="mass-spectrometry.html#loading-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The preferred data structure for use in MSstats is a .csv file in a “long” format with 10 columns representing the following variables: <em>ProteinName, PeptideSequence, PrecursorCharge, FragmentIon, ProductCharge, IsotopeLabelType, Condition, BioReplicate, Run, and Intensity</em>. Note that the variable names are fixed and case-insensitive.</p>
<div class="sourceCode" id="cb239"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb239-1"><a href="mass-spectrometry.html#cb239-1" tabindex="-1"></a><span class="co"># example data provided by the MSstats package</span></span>
<span id="cb239-2"><a href="mass-spectrometry.html#cb239-2" tabindex="-1"></a><span class="fu">head</span>(SRMRawData)</span></code></pre></div>
<pre><code>    ProteinName PeptideSequence PrecursorCharge FragmentIon ProductCharge IsotopeLabelType Condition
243        IDHC   ATDVIVPEEGELR               2          y7            NA                H         1
244        IDHC   ATDVIVPEEGELR               2          y7            NA                L         1
245        IDHC   ATDVIVPEEGELR               2          y8            NA                H         1
246        IDHC   ATDVIVPEEGELR               2          y8            NA                L         1
247        IDHC   ATDVIVPEEGELR               2          y9            NA                H         1
248        IDHC   ATDVIVPEEGELR               2          y9            NA                L         1
    BioReplicate Run   Intensity
243        ReplA   1 84361.08350
244        ReplA   1   215.13526
245        ReplA   1 29778.10188
246        ReplA   1    98.02134
247        ReplA   1 17921.29255
248        ReplA   1    60.47029</code></pre>

</div>
</div>
<div id="statspro-r-package" class="section level3 unnumbered hasAnchor">
<h3>StatsPro R Package<a href="mass-spectrometry.html#statspro-r-package" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><br />
</p>
<table>
<colgroup>
<col width="17%" />
<col width="82%" />
</colgroup>
<tbody>
<tr class="odd">
<td><img src="images/05.png" height="68" /></td>
<td><p><strong>Documentation</strong> | <a href="https://github.com/YanglabWCH/StatsPro#Readme">github</a>, <a href="https://github.com/YanglabWCH/StatsPro/blob/main/StatsPro_Manual.pdf">PDF</a></p>
<p><strong>Code</strong> | <a href="https://github.com/YanglabWCH/StatsPro">github</a></p>
<p><strong>Literature</strong> | <em>Journal of Proteomics</em> 250 (2022): 104386.</p></td>
</tr>
</tbody>
</table>
<p><br />
</p>

</div>
<div id="msgrob2" class="section level3 unnumbered hasAnchor">
<h3>msgrob2<a href="mass-spectrometry.html#msgrob2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><code>msgrob2</code> is an R package that provides functions to perform robust estimation in linear models with missing data. With the help of the Expectation-Maximization (EM) algorithm, the package estimates the parameters of the linear model and imputes the missing data. Additionally, the package offers robust methods for estimating the covariance matrix, including the Minimum Covariance Determinant (MCD) estimator and the S-estimator.</p>
<p>The <code>msgrob2</code> package is particularly useful in situations where data is missing from a linear model. The EM algorithm implemented in the package is a powerful tool for imputing missing data, and the robust covariance estimators allow for a better understanding of the data. The package is designed to provide efficient and accurate results when working with incomplete data, making it an essential tool for researchers and data analysts.</p>
<p><br />
</p>
<table>
<colgroup>
<col width="23%" />
<col width="76%" />
</colgroup>
<tbody>
<tr class="odd">
<td><img src="images/05.png" height="68" /></td>
<td><p><strong>Documentation</strong> | <a href="https://www.bioconductor.org/packages/release/bioc/manuals/msqrob2/man/msqrob2.pdf">PDF</a></p>
<p><strong>Code</strong> | <a href="https://www.bioconductor.org/packages/release/bioc/html/msqrob2.html">Bioconductor</a></p>
<p><strong>Literature</strong> | <em>Molecular &amp; Cellular Proteomics</em>, <strong>15</strong>(2), 657-668.</p>
<p><strong>Literature</strong> | <em>Molecular &amp; Cellular Proteomics</em>, <strong>19</strong>(7), 1209-1219.</p>
<p><strong>Literature</strong> | <em>Analytical Chemistry</em>, <strong>92</strong>(9), 6278–6287.</p></td>
</tr>
</tbody>
</table>
<p><br />
</p>
<div id="installation-4" class="section level4 unnumbered hasAnchor">
<h4>Installation<a href="mass-spectrometry.html#installation-4" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>To install the <code>msgrob2</code> package, you can use the following code:</p>
<div class="sourceCode" id="cb241"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb241-1"><a href="mass-spectrometry.html#cb241-1" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;msgrob2&quot;</span>)</span></code></pre></div>
</div>
<div id="estimating-parameters" class="section level4 unnumbered hasAnchor">
<h4>Estimating Parameters<a href="mass-spectrometry.html#estimating-parameters" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>To estimate the parameters of a linear model with missing data using <code>msgrob2</code>, you can use the <code>msgrob()</code> function. The following code demonstrates how to use <code>msgrob()</code>:</p>
<div class="sourceCode" id="cb242"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb242-1"><a href="mass-spectrometry.html#cb242-1" tabindex="-1"></a><span class="co"># Loading Example Data</span></span>
<span id="cb242-2"><a href="mass-spectrometry.html#cb242-2" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;exMiss&quot;</span>)</span>
<span id="cb242-3"><a href="mass-spectrometry.html#cb242-3" tabindex="-1"></a></span>
<span id="cb242-4"><a href="mass-spectrometry.html#cb242-4" tabindex="-1"></a><span class="co"># Estimating Parameters</span></span>
<span id="cb242-5"><a href="mass-spectrometry.html#cb242-5" tabindex="-1"></a>model <span class="ot">&lt;-</span> <span class="fu">lm</span>(y <span class="sc">~</span> x1 <span class="sc">+</span> x2 <span class="sc">+</span> x3, <span class="at">data =</span> exMiss)</span>
<span id="cb242-6"><a href="mass-spectrometry.html#cb242-6" tabindex="-1"></a><span class="fu">msgrob</span>(model)</span></code></pre></div>
</div>
<div id="estimating-covariance-matrix" class="section level4 unnumbered hasAnchor">
<h4>Estimating Covariance Matrix<a href="mass-spectrometry.html#estimating-covariance-matrix" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>To estimate the covariance matrix using <code>msgrob2</code>, you can use the <code>covrob()</code> function. The following code demonstrates how to use <code>covrob()</code>:</p>
<div class="sourceCode" id="cb243"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb243-1"><a href="mass-spectrometry.html#cb243-1" tabindex="-1"></a><span class="co"># Loading Example Data</span></span>
<span id="cb243-2"><a href="mass-spectrometry.html#cb243-2" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;exMiss&quot;</span>)</span>
<span id="cb243-3"><a href="mass-spectrometry.html#cb243-3" tabindex="-1"></a></span>
<span id="cb243-4"><a href="mass-spectrometry.html#cb243-4" tabindex="-1"></a><span class="co"># Estimating Covariance Matrix</span></span>
<span id="cb243-5"><a href="mass-spectrometry.html#cb243-5" tabindex="-1"></a><span class="fu">covrob</span>(exMiss[,<span class="dv">2</span><span class="sc">:</span><span class="dv">4</span>], <span class="at">method =</span> <span class="st">&quot;MCD&quot;</span>)</span></code></pre></div>
<p>In summary, <code>msgrob2</code> is a comprehensive R package that provides an array of functions for robust estimation in linear models with missing data. The package’s implementation of the EM algorithm and robust covariance estimators make it an essential tool for researchers and data analysts working with incomplete data. The package is easy to install and use, with code examples readily available for reference.</p>

</div>
</div>
<div id="tidyproteomics" class="section level3 unnumbered hasAnchor">
<h3>Tidyproteomics<a href="mass-spectrometry.html#tidyproteomics" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The tidyproteomics R package is a tool that provides a set of functions to preprocess and analyze proteomics data using the tidy data framework. This package is built on top of the tidyverse and Bioconductor packages, which are widely used in the R community for data manipulation and analysis.</p>
<p><br />
</p>
<table>
<colgroup>
<col width="28%" />
<col width="71%" />
</colgroup>
<tbody>
<tr class="odd">
<td><img src="images/05.png" height="68" /></td>
<td><p><strong>Documentation</strong> | <a href="https://jeffsocal.github.io/tidyproteomics/">github</a></p>
<p><strong>Code</strong> | <a href="https://github.com/jeffsocal/tidyproteomics">github</a></p>
<p><strong>Web App</strong> | <a href="bioinformatics.pel.caltech.edu/tidyproteomics/">tidyproteomics</a></p>
<p><strong>Literature</strong></p></td>
</tr>
</tbody>
</table>
<p><br />
</p>
<p>Some of the main features of the tidyproteomics package include:</p>
<ul>
<li>Data preprocessing functions for common tasks such as filtering, normalization, and imputation.</li>
<li>Functions for quality assessment and visualization of proteomics data.</li>
<li>Integration with other Bioconductor packages for downstream analysis such as differential expression analysis and pathway analysis.</li>
</ul>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:10001"></span>
<img src="images/10_001_tidyproteomics.png" alt="\label{fig:10001}tidyproteomics workflow" width="330" />
<p class="caption">
Figure 10.1: tidyproteomics workflow
</p>
</div>
<div id="installation-5" class="section level4 unnumbered hasAnchor">
<h4>Installation<a href="mass-spectrometry.html#installation-5" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>To install the <code>tidyproteomics</code> package, you will need to install GitHub and Bioconductor repositories:</p>
<div class="sourceCode" id="cb244"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb244-1"><a href="mass-spectrometry.html#cb244-1" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;devtools&quot;</span>)</span>
<span id="cb244-2"><a href="mass-spectrometry.html#cb244-2" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;jeffsocal/tidyproteomics&quot;</span>)</span>
<span id="cb244-3"><a href="mass-spectrometry.html#cb244-3" tabindex="-1"></a></span>
<span id="cb244-4"><a href="mass-spectrometry.html#cb244-4" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb244-5"><a href="mass-spectrometry.html#cb244-5" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="fu">c</span>(<span class="st">&quot;limma&quot;</span>,<span class="st">&quot;qvalue&quot;</span>,<span class="st">&quot;fgsea&quot;</span>,<span class="st">&quot;Biostrings&quot;</span>))</span></code></pre></div>
</div>
<div id="loading-data-1" class="section level4 unnumbered hasAnchor">
<h4>Loading Data<a href="mass-spectrometry.html#loading-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>To load your data into tidyproteomics, you can use the following code:</p>
<div class="sourceCode" id="cb245"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb245-1"><a href="mass-spectrometry.html#cb245-1" tabindex="-1"></a><span class="co"># Load the tidyproteomics package</span></span>
<span id="cb245-2"><a href="mass-spectrometry.html#cb245-2" tabindex="-1"></a><span class="fu">library</span>(tidyproteomics)</span>
<span id="cb245-3"><a href="mass-spectrometry.html#cb245-3" tabindex="-1"></a></span>
<span id="cb245-4"><a href="mass-spectrometry.html#cb245-4" tabindex="-1"></a><span class="co"># Import data</span></span>
<span id="cb245-5"><a href="mass-spectrometry.html#cb245-5" tabindex="-1"></a>data_proteins <span class="ot">&lt;-</span> <span class="st">&quot;path_to_data.xlsx&quot;</span> <span class="sc">%&gt;%</span></span>
<span id="cb245-6"><a href="mass-spectrometry.html#cb245-6" tabindex="-1"></a>  <span class="fu">import</span>(<span class="st">&quot;ProteomeDiscoverer&quot;</span>, <span class="st">&quot;proteins&quot;</span>) </span></code></pre></div>
</div>
<div id="data-summaries" class="section level4 unnumbered hasAnchor">
<h4>Data Summaries<a href="mass-spectrometry.html#data-summaries" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Currently, tidyproteomics implements two summary quantitative visualizations. The first is a simple grouped bar chart that displays individual and grouped proteins, as well as all and unique peptides. The match-between-runs is shown as a margin above the MS2 evidenced identifications. In recent literature, a summary of protein quantitation has been visualized as a rank-based dot plot. This plot can be extended to highlight statistical differences via an unbiased all-pair-wise comparison, which gives an anticipated view on how to guide downstream analyses.</p>
<div class="sourceCode" id="cb246"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb246-1"><a href="mass-spectrometry.html#cb246-1" tabindex="-1"></a>p01 <span class="ot">&lt;-</span> data_proteins <span class="sc">%&gt;%</span> <span class="fu">plot_counts</span>()</span>
<span id="cb246-2"><a href="mass-spectrometry.html#cb246-2" tabindex="-1"></a>p02 <span class="ot">&lt;-</span> data_proteins <span class="sc">%&gt;%</span> <span class="fu">plot_quantrank</span>()</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-200-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>Summary Stats {-}
Summarizing proteomics data is vital to understanding the bigger picture and conveying summary stats that set the tone for the larger analysis. The results of each summary can be directed to via the destination option to “print” on screen, “save” to a file or “return” as a tibble.</p>
<div class="sourceCode" id="cb247"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb247-1"><a href="mass-spectrometry.html#cb247-1" tabindex="-1"></a>data_proteins <span class="ot">&lt;-</span> data_proteins <span class="sc">%&gt;%</span></span>
<span id="cb247-2"><a href="mass-spectrometry.html#cb247-2" tabindex="-1"></a>  <span class="co"># save a table of simple summary stats</span></span>
<span id="cb247-3"><a href="mass-spectrometry.html#cb247-3" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="st">&quot;sample&quot;</span>, <span class="at">destination =</span> <span class="st">&quot;save&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb247-4"><a href="mass-spectrometry.html#cb247-4" tabindex="-1"></a>  <span class="co"># save a report on contamination</span></span>
<span id="cb247-5"><a href="mass-spectrometry.html#cb247-5" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">contamination =</span> <span class="st">&quot;CRAP&quot;</span>, <span class="at">destination =</span> <span class="st">&quot;save&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb247-6"><a href="mass-spectrometry.html#cb247-6" tabindex="-1"></a>  <span class="co"># remove contamination</span></span>
<span id="cb247-7"><a href="mass-spectrometry.html#cb247-7" tabindex="-1"></a>  <span class="fu">subset</span>(<span class="sc">!</span>description <span class="sc">%like%</span> <span class="st">&quot;^CRAP&quot;</span>)</span></code></pre></div>
</div>
<div id="normalization-and-imputation" class="section level4 unnumbered hasAnchor">
<h4>Normalization and Imputation<a href="mass-spectrometry.html#normalization-and-imputation" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Quantitative proteomics requires accurate normalization, which can be difficult to implement. The <code>normalize()</code> function in the tidyproteomics package is a wrapper for various normalization methods, while <code>select_normalization()</code> automatically selects the best method based on a weighted score. Both functions allow for downstream analyses such as <code>expression()</code> and <code>enrichment()</code>. The package attempts to apply each function universally to peptide and protein values using the <code>identifier</code> variable to identify the thing being measured.</p>
<div class="sourceCode" id="cb248"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb248-1"><a href="mass-spectrometry.html#cb248-1" tabindex="-1"></a>data_proteins <span class="ot">&lt;-</span> data_proteins <span class="sc">%&gt;%</span></span>
<span id="cb248-2"><a href="mass-spectrometry.html#cb248-2" tabindex="-1"></a>  <span class="co"># normalize via several methods, best method will be automatically selected</span></span>
<span id="cb248-3"><a href="mass-spectrometry.html#cb248-3" tabindex="-1"></a>  <span class="fu">normalize</span>(<span class="at">.method =</span> <span class="fu">c</span>(<span class="st">&quot;median&quot;</span>,<span class="st">&quot;linear&quot;</span>,<span class="st">&quot;limma&quot;</span>,<span class="st">&quot;randomforest&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb248-4"><a href="mass-spectrometry.html#cb248-4" tabindex="-1"></a>  <span class="co"># impute with a minimum value (this is a knock-out)</span></span>
<span id="cb248-5"><a href="mass-spectrometry.html#cb248-5" tabindex="-1"></a>  <span class="fu">impute</span>(base<span class="sc">::</span>min)</span>
<span id="cb248-6"><a href="mass-spectrometry.html#cb248-6" tabindex="-1"></a></span>
<span id="cb248-7"><a href="mass-spectrometry.html#cb248-7" tabindex="-1"></a><span class="co"># plot visualizations comparing normalization methods</span></span>
<span id="cb248-8"><a href="mass-spectrometry.html#cb248-8" tabindex="-1"></a>p03 <span class="ot">&lt;-</span> data_proteins <span class="sc">%&gt;%</span> <span class="fu">plot_normalization</span>()</span>
<span id="cb248-9"><a href="mass-spectrometry.html#cb248-9" tabindex="-1"></a>p04 <span class="ot">&lt;-</span> data_proteins <span class="sc">%&gt;%</span> <span class="fu">plot_variation_cv</span>()</span>
<span id="cb248-10"><a href="mass-spectrometry.html#cb248-10" tabindex="-1"></a>p05 <span class="ot">&lt;-</span> data_proteins <span class="sc">%&gt;%</span> <span class="fu">plot_variation_pca</span>()</span>
<span id="cb248-11"><a href="mass-spectrometry.html#cb248-11" tabindex="-1"></a>p06 <span class="ot">&lt;-</span> data_proteins <span class="sc">%&gt;%</span> <span class="fu">plot_dynamic_range</span>()</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-203-1.png" width="768" style="display: block; margin: auto;" />
<img src="_main_files/figure-html/unnamed-chunk-204-1.png" width="768" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb249"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb249-1"><a href="mass-spectrometry.html#cb249-1" tabindex="-1"></a><span class="co"># plot visualizations of unbiased clustering</span></span>
<span id="cb249-2"><a href="mass-spectrometry.html#cb249-2" tabindex="-1"></a>p07 <span class="ot">&lt;-</span> data_proteins <span class="sc">%&gt;%</span> <span class="fu">plot_heatmap</span>()</span>
<span id="cb249-3"><a href="mass-spectrometry.html#cb249-3" tabindex="-1"></a>p08 <span class="ot">&lt;-</span> data_proteins <span class="sc">%&gt;%</span> <span class="fu">plot_pca</span>()</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-206-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="expression-analysis" class="section level4 unnumbered hasAnchor">
<h4>Expression Analysis<a href="mass-spectrometry.html#expression-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb250"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb250-1"><a href="mass-spectrometry.html#cb250-1" tabindex="-1"></a>data_proteins <span class="ot">&lt;-</span> data_proteins <span class="sc">%&gt;%</span></span>
<span id="cb250-2"><a href="mass-spectrometry.html#cb250-2" tabindex="-1"></a>  <span class="co"># calculate the expression between experiment: ko and control: wt</span></span>
<span id="cb250-3"><a href="mass-spectrometry.html#cb250-3" tabindex="-1"></a>  <span class="fu">expression</span>(kndw<span class="sc">/</span>ctrl) <span class="sc">%&gt;%</span></span>
<span id="cb250-4"><a href="mass-spectrometry.html#cb250-4" tabindex="-1"></a>  <span class="co"># plot the expression analysis</span></span>
<span id="cb250-5"><a href="mass-spectrometry.html#cb250-5" tabindex="-1"></a>  <span class="fu">plot_volcano</span>(kndw<span class="sc">/</span>ctrl, <span class="at">destination =</span> <span class="st">&quot;png&quot;</span>, <span class="at">significance_column =</span> <span class="st">&quot;p_value&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb250-6"><a href="mass-spectrometry.html#cb250-6" tabindex="-1"></a>  <span class="fu">plot_proportion</span>(kndw<span class="sc">/</span>ctrl, <span class="at">destination =</span> <span class="st">&quot;png&quot;</span>)</span></code></pre></div>
<p>Overall, the tidyproteomics package provides a useful set of tools for preprocessing and analyzing proteomics data using the tidy data framework in R. There are several more workable examples in the online documentation.</p>

</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="sharing.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="answers-to-exercises.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/jeffsocal/asms_r_intro/edit/BRANCH/10_01_mass_spectrometry.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
